--- a/components/helium_services/pref_names.h
+++ b/components/helium_services/pref_names.h
@@ -33,6 +33,9 @@ inline constexpr char kHeliumExtProxyEna
 inline constexpr char kHeliumBangsEnabled[] =
     "helium.services.bangs";
 
+inline constexpr char kHeliumUpdateFetchingEnabled[] =
+    "helium.services.browser_updates";
+
 inline constexpr char kHeliumSpellcheckEnabled[] =
     "helium.services.spellcheck_files";
 
--- a/components/helium_services/helium_services_helpers.h
+++ b/components/helium_services/helium_services_helpers.h
@@ -22,11 +22,13 @@ bool ShouldAccessServices(const PrefServ
 
 COMPONENT_EXPORT(HELIUM) bool ShouldFetchBangs(const PrefService& prefs);
 COMPONENT_EXPORT(HELIUM) bool ShouldAccessExtensionService(const PrefService& prefs);
+COMPONENT_EXPORT(HELIUM) bool ShouldAccessUpdateService(const PrefService& prefs);
 COMPONENT_EXPORT(HELIUM) GURL GetServicesBaseURL(const PrefService& prefs);
 COMPONENT_EXPORT(HELIUM) GURL GetDummyURL();
 COMPONENT_EXPORT(HELIUM) GURL GetExtensionUpdateURL(const PrefService& prefs);
 COMPONENT_EXPORT(HELIUM) GURL GetWebstoreSnippetURL(const PrefService& prefs, std::string_view id);
 COMPONENT_EXPORT(HELIUM) GURL GetSpellcheckURL(const PrefService& prefs);
+COMPONENT_EXPORT(HELIUM) GURL GetBrowserUpdateURL(const PrefService& prefs);
 COMPONENT_EXPORT(HELIUM) std::optional<GURL> GetValidUserOverridenURL(std::string_view user_url_);
 COMPONENT_EXPORT(HELIUM) void ConfigurePrefChangeRegistrarFor(std::string_view pref_name,
                             PrefChangeRegistrar& registrar, const base::RepeatingClosure& observer);
--- a/components/helium_services/helium_services_helpers.cc
+++ b/components/helium_services/helium_services_helpers.cc
@@ -6,6 +6,7 @@
 
 #include <optional>
 
+#include "base/command_line.h"
 #include "base/functional/bind.h"
 #include "base/strings/stringprintf.h"
 #include "components/helium_services/pref_names.h"
@@ -15,6 +16,28 @@
 #include "url/url_constants.h"
 
 namespace helium {
+#if defined(ARCH_CPU_ARM64)
+#define CPU_ARCH_STR "arm64"
+#elif defined(ARCH_CPU_X86_64)
+#define CPU_ARCH_STR "x86_64"
+#endif
+
+#if defined(CPU_ARCH_STR) && (BUILDFLAG(IS_MAC) || BUILDFLAG(IS_WIN))
+#define UPDATES_SUPPORTED
+#endif
+
+#if defined(UPDATES_SUPPORTED)
+constexpr std::string_view kUpdateBaseUrl = "https://updates.helium.computer/";
+constexpr const char kUpdatePath[] =
+#if BUILDFLAG(IS_MAC)
+    "mac/appcast-" CPU_ARCH_STR ".xml";
+#elif BUILDFLAG(IS_WIN)
+    "win/appcast.xml";
+#endif
+
+// Duplicated from helium_flag_choices.h to avoid importing it.
+constexpr const char kUpdateServerUrlCommandLine[] = "custom-update-server-url";
+#endif
 
 std::optional<GURL> GetValidUserOverridenURL(std::string_view user_url_) {
     if (user_url_.empty()) {
@@ -73,6 +96,11 @@ bool ShouldFetchBangs(const PrefService&
             prefs.GetBoolean(prefs::kHeliumBangsEnabled);
 }
 
+bool ShouldAccessUpdateService(const PrefService& prefs) {
+  return ShouldAccessServices(prefs) &&
+         prefs.GetBoolean(prefs::kHeliumUpdateFetchingEnabled);
+}
+
 GURL GetExtensionUpdateURL(const PrefService& prefs) {
     if (!ShouldAccessExtensionService(prefs)) {
         return GetDummyURL();
@@ -98,6 +126,24 @@ GURL GetSpellcheckURL(const PrefService&
     return GetServicesBaseURL(prefs).Resolve("/dict/");
 }
 
+GURL GetBrowserUpdateURL(const PrefService& prefs) {
+  if (!ShouldAccessUpdateService(prefs)) {
+    return GetDummyURL();
+  }
+
+#if defined(UPDATES_SUPPORTED)
+  const auto* command_line = base::CommandLine::ForCurrentProcess();
+  std::string baseUrl(kUpdateBaseUrl);
+  if (command_line->HasSwitch(helium::kUpdateServerUrlCommandLine)) {
+    baseUrl =
+        command_line->GetSwitchValueASCII(helium::kUpdateServerUrlCommandLine);
+  }
+  return GURL(baseUrl).Resolve(kUpdatePath);
+#else
+  return GetDummyURL();
+#endif
+}
+
 void ConfigurePrefChangeRegistrarFor(std::string_view pref_name,
     PrefChangeRegistrar& registrar, const base::RepeatingClosure& observer) {
   registrar.Add(prefs::kHeliumServicesEnabled, observer);
--- a/chrome/browser/resources/settings/privacy_page/services_page.html
+++ b/chrome/browser/resources/settings/privacy_page/services_page.html
@@ -99,6 +99,11 @@
                     sub-label="$i18n{heliumSpellcheckToggleDescription}">
                 </settings-toggle-button>
             </if>
+            <settings-toggle-button id="updatesToggleButton"
+                pref="{{prefs.helium.services.browser_updates}}"
+                label="$i18n{heliumUpdatesToggle}"
+                sub-label="$i18n{heliumUpdatesToggleDescription}">
+            </settings-toggle-button>
         </cr-collapse>
     </div>
     </template>
--- a/chrome/browser/ui/webui/settings/settings_localized_strings_provider.cc
+++ b/chrome/browser/ui/webui/settings/settings_localized_strings_provider.cc
@@ -2257,6 +2257,9 @@ void AddPrivacyStrings(content::WebUIDat
       {"heliumSpellcheckToggle", IDS_SETTINGS_HELIUM_SERVICES_SPELLCHECK_TOGGLE},
       {"heliumSpellcheckToggleDescription",
         IDS_SETTINGS_HELIUM_SERVICES_SPELLCHECK_TOGGLE_DESCRIPTION},
+      {"heliumUpdatesToggle", IDS_SETTINGS_HELIUM_SERVICES_UPDATE},
+      {"heliumUpdatesToggleDescription",
+        IDS_SETTINGS_HELIUM_SERVICES_UPDATE_DESCRIPTION},
       {"heliumOriginOverride", IDS_SETTINGS_HELIUM_SERVICES_OVERRIDE},
       {"heliumOriginOverrideDescription",
         IDS_SETTINGS_HELIUM_SERVICES_OVERRIDE_DESCRIPTION},
--- a/chrome/browser/extensions/api/settings_private/prefs_util.cc
+++ b/chrome/browser/extensions/api/settings_private/prefs_util.cc
@@ -406,6 +406,8 @@ const PrefsUtil::TypedPrefMap& PrefsUtil
       settings_api::PrefType::kBoolean;
   (*s_allowlist)[::prefs::kHeliumSpellcheckEnabled] =
       settings_api::PrefType::kBoolean;
+  (*s_allowlist)[::prefs::kHeliumUpdateFetchingEnabled] =
+      settings_api::PrefType::kBoolean;
   (*s_allowlist)[::prefs::kHeliumServicesOrigin] =
       settings_api::PrefType::kString;
   (*s_allowlist)[::prefs::kHeliumServicesConsented] =
--- a/chrome/browser/ui/browser_ui_prefs.cc
+++ b/chrome/browser/ui/browser_ui_prefs.cc
@@ -231,5 +231,6 @@ void RegisterBrowserUserPrefs(user_prefs
     registry->RegisterStringPref(prefs::kHeliumServicesOrigin, "");
     registry->RegisterBooleanPref(prefs::kHeliumDidOnboarding, false);
     registry->RegisterBooleanPref(prefs::kHeliumServicesConsented, false);
+    registry->RegisterBooleanPref(prefs::kHeliumUpdateFetchingEnabled, true);
   }
 
--- a/chrome/app/settings_strings.grdp
+++ b/chrome/app/settings_strings.grdp
@@ -1760,6 +1760,22 @@
     <message name="IDS_SETTINGS_HELIUM_SERVICES_SPELLCHECK_TOGGLE_DESCRIPTION" desc="Description of the for enabling/disabling of downloading spellcheck files">
       Helium will fetch dictionary files used for spell checking when requested. When disabled, spell checking will not work.
     </message>
+    <if expr="not is_linux">
+      <message name="IDS_SETTINGS_HELIUM_SERVICES_UPDATE" desc="Toggle for automatic update downloads">
+        Allow automatic browser updates
+      </message>
+      <message name="IDS_SETTINGS_HELIUM_SERVICES_UPDATE_DESCRIPTION" desc="Description of the toggle for automatic update downloads">
+        Helium will automatically check for updates and install them. This includes browser and component updates. We recommend keeping this setting enabled to ensure you get the latest features and security updates.
+      </message>
+    </if>
+    <if expr="is_linux">
+      <message name="IDS_SETTINGS_HELIUM_SERVICES_UPDATE" desc="Toggle for automatic update downloads">
+        Allow automatic component updates
+      </message>
+      <message name="IDS_SETTINGS_HELIUM_SERVICES_UPDATE_DESCRIPTION" desc="Description of the toggle for automatic update downloads">
+        Helium will automatically check for component updates and install them. We recommend keeping this setting enabled. On Linux, browser updates are handled by your distro's package manager. Please check for Helium package updates at least every week.
+      </message>
+    </if>
     <message name="IDS_SETTINGS_HELIUM_SERVICES_OVERRIDE" desc="Text input for overriding the Helium services server">
       Use your own instance of Helium services
     </message>
--- a/chrome/browser/helium_flag_choices.h
+++ b/chrome/browser/helium_flag_choices.h
@@ -26,6 +26,9 @@ namespace helium {
     {kChannelBeta, kChannelCommandLine, kChannelBeta},
   };
 
+  constexpr const char kUpdateServerUrlCommandLine[] =
+      "custom-update-server-url";
+
 }  // namespace helium
 
 #endif  /* CHROME_BROWSER_HELIUM_FLAG_CHOICES_H_ */
--- a/chrome/browser/helium_flag_entries.h
+++ b/chrome/browser/helium_flag_entries.h
@@ -11,4 +11,7 @@
      "Maximum frame rate for Energy Saver",
      "Configures the frame rate the browser is throttled to when Energy Saver is enabled. Helium flag.",
      kOsDesktop, MULTI_VALUE_TYPE(helium::kEnergySaverFrameRateChoices)},
+    {helium::kUpdateServerUrlCommandLine, "Update server URL",
+     "Overrides the default URL for fetching updates. Should be used only for testing. Helium flag.", kOsDesktop,
+     STRING_VALUE_TYPE(helium::kUpdateServerUrlCommandLine, "")},
 #endif  /* CHROME_BROWSER_HELIUM_FLAG_ENTRIES_H_ */
