--- a/chrome/app/theme/chromium/BRANDING
+++ b/chrome/app/theme/chromium/BRANDING
@@ -1,10 +1,10 @@
-COMPANY_FULLNAME=The Chromium Authors
-COMPANY_SHORTNAME=The Chromium Authors
-PRODUCT_FULLNAME=Chromium
-PRODUCT_SHORTNAME=Chromium
-PRODUCT_INSTALLER_FULLNAME=Chromium Installer
-PRODUCT_INSTALLER_SHORTNAME=Chromium Installer
-COPYRIGHT=Copyright @LASTCHANGE_YEAR@ The Chromium Authors. All rights reserved.
-MAC_BUNDLE_ID=org.chromium.Chromium
+COMPANY_FULLNAME=The Nitrous Authors
+COMPANY_SHORTNAME=The Nitrous Authors
+PRODUCT_FULLNAME=Nitrous
+PRODUCT_SHORTNAME=Nitrous
+PRODUCT_INSTALLER_FULLNAME=Nitrous Installer
+PRODUCT_INSTALLER_SHORTNAME=Nitrous Installer
+COPYRIGHT=Copyright @LASTCHANGE_YEAR@ The Nitrous Authors. All rights reserved.
+MAC_BUNDLE_ID=net.imput.nitrous
 MAC_CREATOR_CODE=Cr24
-MAC_TEAM_ID=
+MAC_TEAM_ID=S4Q33XPHB4
