--- a/chrome/common/pref_names.h
+++ b/chrome/common/pref_names.h
@@ -738,6 +738,11 @@ inline constexpr char
         "restricted_managed_guest_session_extension_cleanup_exempt_list";
 #endif  // BUILDFLAG(IS_CHROMEOS)
 
+// A boolean pref set to true if new tabs should be created to the next of
+// current active one, instead of tab strip end.
+inline constexpr char kNewTabNextToActive[] =
+    "helium.browser.new_tab_next_to_active";
+
 // A boolean pref set to true if a Home button to open the Home pages should be
 // visible on the toolbar.
 inline constexpr char kShowHomeButton[] = "browser.show_home_button";
--- a/chrome/browser/ui/browser_ui_prefs.cc
+++ b/chrome/browser/ui/browser_ui_prefs.cc
@@ -123,4 +123,6 @@ void RegisterBrowserUserPrefs(user_prefs
   registry->RegisterBooleanPref(prefs::kHomePageIsNewTabPage, true,
                                 pref_registration_flags);
+  registry->RegisterBooleanPref(prefs::kNewTabNextToActive, false);
+
   registry->RegisterBooleanPref(prefs::kShowHomeButton, false,
                                 pref_registration_flags);
--- a/chrome/browser/ui/tabs/tab_strip_model.cc
+++ b/chrome/browser/ui/tabs/tab_strip_model.cc
@@ -84,12 +84,14 @@
 #include "chrome/browser/web_applications/web_app_provider.h"
 #include "chrome/browser/web_applications/web_app_tab_helper.h"
 #include "chrome/common/chrome_features.h"
+#include "chrome/common/pref_names.h"
 #include "chrome/common/webui_url_constants.h"
 #include "components/commerce/core/commerce_utils.h"
 #include "components/content_settings/core/browser/host_content_settings_map.h"
 #include "components/content_settings/core/common/content_settings.h"
 #include "components/content_settings/core/common/content_settings_types.h"
 #include "components/feature_engagement/public/feature_constants.h"
+#include "components/prefs/pref_service.h"
 #include "components/reading_list/core/reading_list_model.h"
 #include "components/split_tabs/split_tab_id.h"
 #include "components/split_tabs/split_tab_visual_data.h"
@@ -1753,7 +1755,20 @@ void TabStripModel::AddTab(std::unique_p
     // For all other types, respect what was passed to us, normalizing -1s and
     // values that are too large.
     if (index < 0 || index > count()) {
-      index = count();
+      const bool new_tab_next_to_active =
+          profile_->GetPrefs()->GetBoolean(prefs::kNewTabNextToActive);
+      const auto opener_group = GetTabGroupForTab(active_index());
+
+      if (count() >= 1 && new_tab_next_to_active) {
+        if (group_model_ && opener_group && !(add_types & ADD_ACTIVE)) {
+          index = static_cast<int>(
+              group_model_->GetTabGroup(*opener_group)->ListTabs().end());
+        } else {
+          index = active_index() + 1;
+        }
+      } else {
+        index = count();
+      }
     }
   }
 
--- a/chrome/app/settings_strings.grdp
+++ b/chrome/app/settings_strings.grdp
@@ -261,6 +261,9 @@
     <message name="IDS_SETTINGS_COLOR_SCHEME_MODE" desc="Text of the label for selecting a color scheme mode for the browser.">
       Mode
     </message>
+    <message name="IDS_SETTINGS_NEW_TAB_NEXT_TO_ACTIVE_BUTTON" desc="Label for the checkbox which enables or disables behaviour of opening new tabs next to current active tab.">
+      Open new tabs next to the active tab
+    </message>
     <message name="IDS_SETTINGS_SHOW_HOME_BUTTON" desc="Label for the checkbox which enables or disables showing the home button in the toolbar.">
       Show home button
     </message>
--- a/chrome/browser/ui/webui/settings/settings_localized_strings_provider.cc
+++ b/chrome/browser/ui/webui/settings/settings_localized_strings_provider.cc
@@ -513,6 +513,7 @@ void AddAppearanceStrings(content::WebUI
       {"lightMode", IDS_NTP_CUSTOMIZE_CHROME_COLOR_SCHEME_MODE_LIGHT_LABEL},
       {"darkMode", IDS_NTP_CUSTOMIZE_CHROME_COLOR_SCHEME_MODE_DARK_LABEL},
       {"systemMode", IDS_NTP_CUSTOMIZE_CHROME_COLOR_SCHEME_MODE_SYSTEM_LABEL},
+      {"openNewTabNextToActive", IDS_SETTINGS_NEW_TAB_NEXT_TO_ACTIVE_BUTTON},
       {"showHomeButton", IDS_SETTINGS_SHOW_HOME_BUTTON},
       {"showBookmarksBar", IDS_SETTINGS_SHOW_BOOKMARKS_BAR},
       {"showTabSearchButton", IDS_SETTINGS_SHOW_TAB_SEARCH_BUTTON},
--- a/chrome/browser/extensions/api/settings_private/prefs_util.cc
+++ b/chrome/browser/extensions/api/settings_private/prefs_util.cc
@@ -262,6 +262,10 @@ const PrefsUtil::TypedPrefMap& PrefsUtil
   (*s_allowlist)[::prefs::kUseCustomChromeFrame] =
       settings_api::PrefType::kBoolean;
 #endif
+
+  (*s_allowlist)[::prefs::kNewTabNextToActive] =
+      settings_api::PrefType::kBoolean;
+
   (*s_allowlist)[::prefs::kShowHomeButton] = settings_api::PrefType::kBoolean;
   (*s_allowlist)[::prefs::kShowForwardButton] =
       settings_api::PrefType::kBoolean;
--- a/chrome/browser/resources/settings/appearance_page/appearance_behavior_page.html
+++ b/chrome/browser/resources/settings/appearance_page/appearance_behavior_page.html
@@ -1,6 +1,11 @@
 <style include="cr-shared-style settings-shared"></style>
 <settings-section page-title="$i18n{behaviorPageTitle}">
 
+  <settings-toggle-button id="newTabNextToActive"
+      pref="{{prefs.helium.browser.new_tab_next_to_active}}"
+      label="$i18n{openNewTabNextToActive}">
+  </settings-toggle-button>
+
   <settings-toggle-button class="hr" id="copyPageUrlShortcut"
       pref="{{prefs.helium.settings.a11y.copy_page_url_shortcut}}"
       label="$i18n{copyPageUrlShortcut}">
