diff --git a/chrome/browser/resources/settings/privacy_page/persona_page.ts b/chrome/browser/resources/settings/privacy_page/persona_page.ts
index 75e39f5..2328696 100644
--- a/chrome/browser/resources/settings/privacy_page/persona_page.ts
+++ b/chrome/browser/resources/settings/privacy_page/persona_page.ts
@@ -1346,7 +1346,7 @@ export class SettingsHeliumPersonaPageElement extends PolymerElement {
   }
 
   private stringListFromInput_(value: string): string[] {
-    return value.split(',')
+    return value.split(/[\r\n,]+/)
         .map(item => item.trim())
         .filter(item => item.length > 0);
   }
@@ -1367,7 +1367,7 @@ export class SettingsHeliumPersonaPageElement extends PolymerElement {
   }
 
   private uaChBrandsFromInput_(value: string): Array<Record<string, string>> {
-    return value.split(',')
+    return value.split(/[\r\n,]+/)
         .map(item => item.trim())
         .filter(Boolean)
         .map(item => {
