diff --git a/chrome/browser/resources/settings/privacy_page/persona_page.ts b/chrome/browser/resources/settings/privacy_page/persona_page.ts
index 46d5074..9c91fdf 100644
--- a/chrome/browser/resources/settings/privacy_page/persona_page.ts
+++ b/chrome/browser/resources/settings/privacy_page/persona_page.ts
@@ -2376,10 +2376,10 @@ export class SettingsHeliumPersonaPageElement extends PolymerElement {
     persona.modified = true;
     persona.passthrough = false;
     persona.noiseSeed = this.makeRandomNoiseSeed_();
-    this.applyRandomUaVersion_(persona);
     this.applyRandomRegion_(persona);
     this.applyRandomHardware_(persona);
     this.applyRandomGpu_(persona);
+    this.applyRandomUaVersion_(persona);
     this.applyRandomScreen_(persona);
     this.applyConsistentRandomDeviceLabel_(persona);
     this.applyRandomDisplayPreferences_(persona);
