diff --git a/chrome/browser/resources/settings/privacy_page/persona_page.ts b/chrome/browser/resources/settings/privacy_page/persona_page.ts
index cddfbb40..e8fc30db 100644
--- a/chrome/browser/resources/settings/privacy_page/persona_page.ts
+++ b/chrome/browser/resources/settings/privacy_page/persona_page.ts
@@ -1339,23 +1339,23 @@ export class SettingsHeliumPersonaPageElement extends PolymerElement {
     ];
     const macProfiles = [
       this.makeGpuProfile_(
-          'Apple Inc.', 'Apple M1', 'Apple M1', 'Metal',
+          'Apple Inc.', 'ANGLE (Apple, Apple M1, Metal)', 'Apple M1', 'Metal',
           16384, 4, 16, 268435456, []),
       this.makeGpuProfile_(
-          'Apple Inc.', 'Apple M2', 'Apple M2', 'Metal',
+          'Apple Inc.', 'ANGLE (Apple, Apple M2, Metal)', 'Apple M2', 'Metal',
           16384, 4, 16, 268435456, []),
       this.makeGpuProfile_(
-          'Apple Inc.', 'Apple M3', 'Apple M3', 'Metal',
+          'Apple Inc.', 'ANGLE (Apple, Apple M3, Metal)', 'Apple M3', 'Metal',
           16384, 4, 16, 536870912, []),
     ];
     const linuxProfiles = [
       this.makeGpuProfile_(
-          'Intel Inc.',
+          'Mesa',
           'Mesa Intel(R) UHD Graphics 620 (KBL GT2)',
           'Intel UHD Graphics 620', 'Mesa', 16384, 4, 16, 268435456,
           ['texture-compression-bc']),
       this.makeGpuProfile_(
-          'AMD',
+          'Mesa',
           'AMD Radeon RX 580 Series (RADV POLARIS10)',
           'AMD Radeon RX 580', 'Mesa RADV', 16384, 8, 16, 268435456,
           ['texture-compression-bc']),
