--- a/third_party/blink/renderer/core/loader/frame_fetch_context.cc
+++ b/third_party/blink/renderer/core/loader/frame_fetch_context.cc
@@ -685,6 +685,10 @@
   using network::mojom::blink::WebClientHintsType;
 
   ExecutionContext* execution_context = GetExecutionContext();
+  if (PersonaSpoofingHelper::IsEnabled(execution_context) &&
+      !PersonaSpoofingHelper::ClientHintsEnabled(execution_context)) {
+    return;
+  }
   const std::optional<float> persona_device_memory =
       PersonaSpoofingHelper::DeviceMemory(execution_context);
 
