--- a/third_party/blink/renderer/core/execution_context/navigator_base.cc
+++ b/third_party/blink/renderer/core/execution_context/navigator_base.cc
@@ -14,5 +15,6 @@
 #include "third_party/blink/renderer/platform/runtime_enabled_features.h"
 #include "third_party/blink/renderer/platform/wtf/text/string_builder.h"
+#include "third_party/blink/renderer/platform/wtf/text/string_view.h"
 
 #if !BUILDFLAG(IS_MAC) && !BUILDFLAG(IS_WIN)
 #include <sys/utsname.h>
@@ -110,7 +115,13 @@ String NavigatorBase::platform() const {
 #endif
   return GetReducedNavigatorPlatform();
 }
 
+bool NavigatorBase::onLine() {
+  if (PersonaSpoofingHelper::IsEnabled(GetExecutionContext()))
+    return PersonaSpoofingHelper::NetworkOnline(GetExecutionContext());
+  return NavigatorOnLine::onLine();
+}
+
 void NavigatorBase::Trace(Visitor* visitor) const {
   ScriptWrappable::Trace(visitor);
   NavigatorLanguage::Trace(visitor);
--- a/third_party/blink/renderer/core/execution_context/navigator_base.h
+++ b/third_party/blink/renderer/core/execution_context/navigator_base.h
@@ -57,6 +52,7 @@ class CORE_EXPORT NavigatorBase : public ScriptWrappable,
   // NavigatorID override
   String userAgent() const override;
   String platform() const override;
+  bool onLine() override;
   void Trace(Visitor* visitor) const override;
 
   unsigned int hardwareConcurrency() const override;
--- a/third_party/blink/renderer/core/frame/navigator_on_line.h
+++ b/third_party/blink/renderer/core/frame/navigator_on_line.h
@@ -38,7 +38,7 @@ namespace blink {
 
 class CORE_EXPORT NavigatorOnLine {
  public:
-  bool onLine() { return GetNetworkStateNotifier().OnLine(); }
+  virtual bool onLine() { return GetNetworkStateNotifier().OnLine(); }
 };
 
 }  // namespace blink
