--- a/content/browser/renderer_host/render_frame_host_impl.cc
+++ b/content/browser/renderer_host/render_frame_host_impl.cc
@@ -15005,6 +15005,12 @@
 
 void RenderFrameHostImpl::GetWebAuthenticationService(
     mojo::PendingReceiver<blink::mojom::Authenticator> receiver) {
+  const blink::HeliumPersonaSnapshot& persona_snapshot =
+      GetPage().persona_snapshot();
+  if (persona_snapshot.IsEnabled() &&
+      !persona_snapshot.allow_platform_credentials) {
+    return;
+  }
 #if !BUILDFLAG(IS_ANDROID)
   AuthenticatorImpl::Create(this, std::move(receiver));
 #else
