--- a/content/browser/web_contents/web_contents_impl.cc
+++ b/content/browser/web_contents/web_contents_impl.cc
@@ -87,6 +87,7 @@
 #include "content/browser/find_request_manager.h"
 #include "content/browser/gpu/gpu_data_manager_impl.h"
 #include "content/browser/gpu/gpu_process_host.h"
+#include "content/browser/helium_noise/noise_token_data.h"
 #include "content/browser/guest_page_holder_impl.h"
 #include "content/browser/host_zoom_map_impl.h"
 #include "content/browser/media/audio_stream_monitor.h"
@@ -2123,13 +2124,26 @@ std::optional<SkColor> WebContentsImpl::GetBackgroundColor() {
 void WebContentsImpl::UpdateHeliumPersonaSnapshot(
     const blink::HeliumPersonaSnapshot& snapshot) {
   ForEachFrameTree([&snapshot](FrameTree& frame_tree) {
-    frame_tree.root()->current_frame_host()->GetPage().set_persona_snapshot(
-        snapshot);
+    RenderFrameHostImpl* current_frame_host =
+        frame_tree.root()->current_frame_host();
+    PageImpl& page = current_frame_host->GetPage();
+    page.set_persona_snapshot(snapshot);
+
+    BrowserContext* browser_context = current_frame_host->GetBrowserContext();
+    const url::Origin origin = current_frame_host->GetLastCommittedOrigin();
+    blink::HeliumNoiseTokenMap noise_tokens;
+    if (GetContentClient()->browser()->HeliumNoiseEnabled(browser_context,
+                                                          origin.GetURL())) {
+      noise_tokens =
+          HeliumNoiseTokenData::GetTokens(browser_context, origin, snapshot);
+    }
+    page.set_noise_tokens(noise_tokens);
     frame_tree.root()->render_manager()->ExecutePageBroadcastMethod(
-        [&snapshot](RenderViewHostImpl* rvh) {
+        [&snapshot, noise_tokens](RenderViewHostImpl* rvh) {
           if (auto& broadcast = rvh->GetAssociatedPageBroadcast()) {
+            broadcast->UpdateHeliumNoiseTokens(noise_tokens);
             broadcast->UpdateHeliumPersonaSnapshot(snapshot);
           }
         });
   });
 }
