diff --git a/third_party/blink/renderer/core/loader/frame_fetch_context.cc b/third_party/blink/renderer/core/loader/frame_fetch_context.cc
index 4701626..99f05c6 100644
--- a/third_party/blink/renderer/core/loader/frame_fetch_context.cc
+++ b/third_party/blink/renderer/core/loader/frame_fetch_context.cc
@@ -924,8 +924,12 @@ void FrameFetchContext::AddClientHintsIfNecessary(
   }
 
   if (LocalFrameView* frame_view = GetFrame()->View()) {
-    const int viewport_width = frame_view->ViewportWidth();
-    const int viewport_height = frame_view->ViewportHeight();
+    const int viewport_width =
+        PersonaSpoofingHelper::ScreenWidth(execution_context)
+            .value_or(frame_view->ViewportWidth());
+    const int viewport_height =
+        PersonaSpoofingHelper::ScreenHeight(execution_context)
+            .value_or(frame_view->ViewportHeight());
     if (ShouldSendClientHint(*policy, resource_origin, is_1p_origin,
                              WebClientHintsType::kViewportWidth_DEPRECATED,
                              hints_preferences)) {
