diff --git a/third_party/blink/renderer/modules/webgpu/gpu_adapter.cc b/third_party/blink/renderer/modules/webgpu/gpu_adapter.cc
index 58f2ea7..ba44a56 100644
--- a/third_party/blink/renderer/modules/webgpu/gpu_adapter.cc
+++ b/third_party/blink/renderer/modules/webgpu/gpu_adapter.cc
@@ -34,32 +34,6 @@ namespace blink {
 
 namespace {
 
-std::optional<V8GPUFeatureName::Enum> PersonaWebGPUFeatureEnum(
-    const String& feature) {
-  if (DeprecatedEqualIgnoringCase(feature, "depth32float-stencil8")) {
-    return V8GPUFeatureName::Enum::kDepth32FloatStencil8;
-  }
-  if (DeprecatedEqualIgnoringCase(feature, "texture-compression-bc")) {
-    return V8GPUFeatureName::Enum::kTextureCompressionBc;
-  }
-  if (DeprecatedEqualIgnoringCase(feature, "texture-compression-etc2")) {
-    return V8GPUFeatureName::Enum::kTextureCompressionEtc2;
-  }
-  if (DeprecatedEqualIgnoringCase(feature, "texture-compression-astc")) {
-    return V8GPUFeatureName::Enum::kTextureCompressionAstc;
-  }
-  if (DeprecatedEqualIgnoringCase(feature, "timestamp-query")) {
-    return V8GPUFeatureName::Enum::kTimestampQuery;
-  }
-  if (DeprecatedEqualIgnoringCase(feature, "texture-formats-tier1")) {
-    return V8GPUFeatureName::Enum::kTextureFormatsTier1;
-  }
-  if (DeprecatedEqualIgnoringCase(feature, "dual-source-blending")) {
-    return V8GPUFeatureName::Enum::kDualSourceBlending;
-  }
-  return std::nullopt;
-}
-
 GPUSupportedFeatures* MakePersonaFeatureNameSet(
     const ExecutionContext* execution_context) {
   GPUSupportedFeatures* features = MakeGarbageCollected<GPUSupportedFeatures>();
diff --git a/third_party/blink/renderer/modules/webgpu/gpu_device.cc b/third_party/blink/renderer/modules/webgpu/gpu_device.cc
index 95829a0..a3013fc 100644
--- a/third_party/blink/renderer/modules/webgpu/gpu_device.cc
+++ b/third_party/blink/renderer/modules/webgpu/gpu_device.cc
@@ -55,32 +55,6 @@ namespace blink {
 
 namespace {
 
-std::optional<V8GPUFeatureName::Enum> PersonaWebGPUFeatureEnum(
-    const String& feature) {
-  if (DeprecatedEqualIgnoringCase(feature, "depth32float-stencil8")) {
-    return V8GPUFeatureName::Enum::kDepth32FloatStencil8;
-  }
-  if (DeprecatedEqualIgnoringCase(feature, "texture-compression-bc")) {
-    return V8GPUFeatureName::Enum::kTextureCompressionBc;
-  }
-  if (DeprecatedEqualIgnoringCase(feature, "texture-compression-etc2")) {
-    return V8GPUFeatureName::Enum::kTextureCompressionEtc2;
-  }
-  if (DeprecatedEqualIgnoringCase(feature, "texture-compression-astc")) {
-    return V8GPUFeatureName::Enum::kTextureCompressionAstc;
-  }
-  if (DeprecatedEqualIgnoringCase(feature, "timestamp-query")) {
-    return V8GPUFeatureName::Enum::kTimestampQuery;
-  }
-  if (DeprecatedEqualIgnoringCase(feature, "texture-formats-tier1")) {
-    return V8GPUFeatureName::Enum::kTextureFormatsTier1;
-  }
-  if (DeprecatedEqualIgnoringCase(feature, "dual-source-blending")) {
-    return V8GPUFeatureName::Enum::kDualSourceBlending;
-  }
-  return std::nullopt;
-}
-
 GPUSupportedFeatures* MakePersonaFeatureNameSet(
     const ExecutionContext* execution_context) {
   GPUSupportedFeatures* features = MakeGarbageCollected<GPUSupportedFeatures>();
