--- a/third_party/blink/renderer/modules/screen_orientation/DEPS
+++ b/third_party/blink/renderer/modules/screen_orientation/DEPS
@@ -1,4 +1,5 @@
 include_rules = [
+    "+third_party/blink/renderer/core/helium_persona",
     "+services/device/public/mojom/screen_orientation.mojom-blink.h",
     "+services/device/public/mojom/screen_orientation.mojom-blink-forward.h",
     "-third_party/blink/renderer/modules",
--- a/third_party/blink/renderer/modules/screen_orientation/screen_orientation.cc
+++ b/third_party/blink/renderer/modules/screen_orientation/screen_orientation.cc
@@ -13,6 +13,7 @@
 #include "third_party/blink/renderer/bindings/modules/v8/v8_orientation_lock_type.h"
 #include "third_party/blink/renderer/core/dom/dom_exception.h"
 #include "third_party/blink/renderer/core/frame/local_dom_window.h"
+#include "third_party/blink/renderer/core/helium_persona/persona_spoofing_helper.h"
 #include "third_party/blink/renderer/modules/event_target_modules.h"
 #include "third_party/blink/renderer/modules/screen_orientation/lock_orientation_callback.h"
 #include "third_party/blink/renderer/modules/screen_orientation/screen_orientation_controller.h"
@@ -88,10 +89,16 @@
 }
 
 V8OrientationType ScreenOrientation::type() const {
+  if (PersonaSpoofingHelper::IsEnabled(GetExecutionContext())) {
+    return V8OrientationType(V8OrientationType::Enum::kLandscapePrimary);
+  }
   return V8OrientationType(OrientationTypeToV8Enum(type_));
 }
 
 uint16_t ScreenOrientation::angle() const {
+  if (PersonaSpoofingHelper::IsEnabled(GetExecutionContext())) {
+    return 0;
+  }
   return angle_;
 }
 
