diff --git a/content/browser/client_hints/client_hints.cc b/content/browser/client_hints/client_hints.cc
index cfeb929e..bddd4a3b 100644
--- a/content/browser/client_hints/client_hints.cc
+++ b/content/browser/client_hints/client_hints.cc
@@ -526,15 +526,15 @@ void AddPersonaDeviceMemoryHeader(
   const WebClientHintsType header =
       use_deprecated_version ? WebClientHintsType::kDeviceMemory_DEPRECATED
                              : WebClientHintsType::kDeviceMemory;
+  if (persona_snapshot.reduce_system_info) {
+    SetHeaderToDouble(headers, header, 2);
+    return;
+  }
   if (persona_snapshot.device_memory > 0) {
     SetHeaderToDouble(headers, header,
                       DeviceMemoryBucket(persona_snapshot.device_memory));
     return;
   }
-  if (persona_snapshot.reduce_system_info) {
-    SetHeaderToDouble(headers, header, 2);
-    return;
-  }
   if (persona_snapshot.device_memory <= 0) {
     AddDeviceMemoryHeader(headers, use_deprecated_version);
     return;
diff --git a/third_party/blink/renderer/core/loader/worker_fetch_context.cc b/third_party/blink/renderer/core/loader/worker_fetch_context.cc
index b44b6281..6804cedf 100644
--- a/third_party/blink/renderer/core/loader/worker_fetch_context.cc
+++ b/third_party/blink/renderer/core/loader/worker_fetch_context.cc
@@ -4,6 +4,7 @@
 
 #include "third_party/blink/renderer/core/loader/worker_fetch_context.h"
 
+#include "base/strings/string_util.h"
 #include "base/task/single_thread_task_runner.h"
 #include "third_party/blink/public/mojom/fetch/fetch_api_request.mojom-blink.h"
 #include "third_party/blink/public/platform/web_content_settings_client.h"
@@ -28,6 +29,7 @@
 #include "third_party/blink/renderer/platform/loader/fetch/worker_resource_timing_notifier.h"
 #include "third_party/blink/renderer/platform/network/http_names.h"
 #include "third_party/blink/renderer/platform/network/network_state_notifier.h"
+#include "third_party/blink/renderer/platform/network/network_utils.h"
 #include "third_party/blink/renderer/platform/runtime_enabled_features.h"
 #include "third_party/blink/renderer/platform/scheduler/public/virtual_time_controller.h"
 #include "third_party/blink/renderer/platform/supplementable.h"
@@ -231,11 +233,22 @@ void WorkerFetchContext::AddAdditionalRequestHeaders(ResourceRequest& request) {
 
   if (const HeliumPersonaSnapshot* snapshot =
           PersonaSpoofingHelper::Get(global_scope_.Get())) {
-    if (!snapshot->accept_language.empty() &&
-        request.HttpHeaderField(http_names::kAcceptLanguage).empty()) {
-      request.SetHttpHeaderField(
-          http_names::kAcceptLanguage,
-          AtomicString(String::FromUtf8(snapshot->accept_language)));
+    if (request.HttpHeaderField(http_names::kAcceptLanguage).empty()) {
+      String accept_language;
+      if (!snapshot->accept_language.empty()) {
+        accept_language = String::FromUtf8(snapshot->accept_language);
+      } else if (!snapshot->navigator_languages.empty()) {
+        accept_language = network_utils::GenerateAcceptLanguageHeader(
+            String::FromUtf8(
+                base::JoinString(snapshot->navigator_languages, ",")));
+      } else if (!snapshot->locale.empty()) {
+        accept_language = network_utils::GenerateAcceptLanguageHeader(
+            String::FromUtf8(snapshot->locale));
+      }
+      if (!accept_language.empty()) {
+        request.SetHttpHeaderField(http_names::kAcceptLanguage,
+                                   AtomicString(accept_language));
+      }
     }
   }
 
