diff --git a/third_party/blink/renderer/core/loader/worker_fetch_context.cc b/third_party/blink/renderer/core/loader/worker_fetch_context.cc
index 2558ee3..10ac895 100644
--- a/third_party/blink/renderer/core/loader/worker_fetch_context.cc
+++ b/third_party/blink/renderer/core/loader/worker_fetch_context.cc
@@ -224,11 +224,6 @@ void WorkerFetchContext::AddAdditionalRequestHeaders(ResourceRequest& request) {
     return;
   }
 
-  if (PersonaSpoofingHelper::IsEnabled(global_scope_.Get()) &&
-      !PersonaSpoofingHelper::ClientHintsEnabled(global_scope_.Get())) {
-    return;
-  }
-
   if (const HeliumPersonaSnapshot* snapshot =
           PersonaSpoofingHelper::Get(global_scope_.Get())) {
     if (!snapshot->accept_language.empty() &&
