--- a/chrome/browser/spellchecker/spellcheck_hunspell_dictionary.cc
+++ b/chrome/browser/spellchecker/spellcheck_hunspell_dictionary.cc
@@ -24,8 +24,10 @@
 #include "base/task/thread_pool.h"
 #include "base/threading/scoped_blocking_call.h"
 #include "build/build_config.h"
+#include "chrome/browser/profiles/profile.h"
 #include "chrome/browser/spellchecker/spellcheck_service.h"
 #include "chrome/common/chrome_paths.h"
+#include "components/helium_services/helium_services_helpers.h"
 #include "components/spellcheck/browser/spellcheck_platform.h"
 #include "components/spellcheck/common/spellcheck_common.h"
 #include "components/spellcheck/common/spellcheck_features.h"
@@ -277,15 +279,26 @@ GURL SpellcheckHunspellDictionary::GetDi
   std::string bdict_file = dictionary_file_.path.BaseName().MaybeAsASCII();
   DCHECK(!bdict_file.empty());
 
-  static const char kDownloadServerUrl[] =
-      "https://redirector.gvt1.com/edgedl/chrome/dict/";
+  if (browser_context_) {
+    const Profile* profile = Profile::FromBrowserContext(browser_context_);
+    if (!profile) {
+      goto fail;
+    }
+
+    const PrefService* prefs = profile->GetPrefs();
+    if (!prefs) {
+      goto fail;
+    }
+
+    return helium::GetSpellcheckURL(*prefs)
+            .Resolve(bdict_file);
+  }
 
-  return GURL(std::string(kDownloadServerUrl) +
-              base::ToLowerASCII(bdict_file));
+fail:
+  return GURL("https://failed-getting-spellcheck.qjz9zk/");
 }
 
 void SpellcheckHunspellDictionary::DownloadDictionary(GURL url) {
-  return;
   DCHECK_CURRENTLY_ON(BrowserThread::UI);
   DCHECK(browser_context_);
 
--- a/components/helium_services/helium_services_helpers.cc
+++ b/components/helium_services/helium_services_helpers.cc
@@ -63,6 +63,11 @@ bool ShouldAccessExtensionService(const
             prefs.GetBoolean(prefs::kHeliumExtProxyEnabled);
 }
 
+bool ShouldAccessDictionaryService(const PrefService& prefs) {
+    return ShouldAccessServices(prefs) &&
+            prefs.GetBoolean(prefs::kHeliumSpellcheckEnabled);
+}
+
 bool ShouldFetchBangs(const PrefService& prefs) {
     return ShouldAccessServices(prefs) &&
             prefs.GetBoolean(prefs::kHeliumBangsEnabled);
@@ -85,6 +90,14 @@ GURL GetWebstoreSnippetURL(const PrefSer
         base::StringPrintf("/ext/cws_snippet?id=%s", id));
 }
 
+GURL GetSpellcheckURL(const PrefService& prefs) {
+    if (!ShouldAccessDictionaryService(prefs)) {
+        return GetDummyURL();
+    }
+
+    return GetServicesBaseURL(prefs).Resolve("/dict/");
+}
+
 void ConfigurePrefChangeRegistrarFor(std::string_view pref_name,
     PrefChangeRegistrar& registrar, const base::RepeatingClosure& observer) {
   registrar.Add(prefs::kHeliumServicesEnabled, observer);
--- a/components/helium_services/helium_services_helpers.h
+++ b/components/helium_services/helium_services_helpers.h
@@ -26,6 +26,7 @@ COMPONENT_EXPORT(HELIUM) GURL GetService
 COMPONENT_EXPORT(HELIUM) GURL GetDummyURL();
 COMPONENT_EXPORT(HELIUM) GURL GetExtensionUpdateURL(const PrefService& prefs);
 COMPONENT_EXPORT(HELIUM) GURL GetWebstoreSnippetURL(const PrefService& prefs, std::string_view id);
+COMPONENT_EXPORT(HELIUM) GURL GetSpellcheckURL(const PrefService& prefs);
 COMPONENT_EXPORT(HELIUM) std::optional<GURL> GetValidUserOverridenURL(std::string_view user_url_);
 COMPONENT_EXPORT(HELIUM) void ConfigurePrefChangeRegistrarFor(std::string_view pref_name,
                             PrefChangeRegistrar& registrar, const base::RepeatingClosure& observer);
--- a/components/helium_services/pref_names.h
+++ b/components/helium_services/pref_names.h
@@ -33,6 +33,9 @@ inline constexpr char kHeliumExtProxyEna
 inline constexpr char kHeliumBangsEnabled[] =
     "helium.services.bangs";
 
+inline constexpr char kHeliumSpellcheckEnabled[] =
+    "helium.services.spellcheck_files";
+
 }  // namespace prefs
 
 #endif  // COMPONENTS_HELIUM_SERVICES_PREF_NAMES_H_
--- a/chrome/browser/BUILD.gn
+++ b/chrome/browser/BUILD.gn
@@ -5656,6 +5656,8 @@ static_library("test_support") {
     deps += [
       "//chrome/browser/spellchecker",
       "//components/spellcheck/common",
+      "//components/helium_services",
+      "//chrome/browser/profiles:profile",
     ]
   }
   if (enable_bound_session_credentials) {
--- a/chrome/browser/extensions/api/settings_private/prefs_util.cc
+++ b/chrome/browser/extensions/api/settings_private/prefs_util.cc
@@ -404,6 +404,8 @@ const PrefsUtil::TypedPrefMap& PrefsUtil
       settings_api::PrefType::kBoolean;
   (*s_allowlist)[::prefs::kHeliumBangsEnabled] =
       settings_api::PrefType::kBoolean;
+  (*s_allowlist)[::prefs::kHeliumSpellcheckEnabled] =
+      settings_api::PrefType::kBoolean;
   (*s_allowlist)[::prefs::kHeliumServicesOrigin] =
       settings_api::PrefType::kString;
   (*s_allowlist)[::prefs::kHeliumServicesConsented] =
--- a/chrome/browser/ui/browser_ui_prefs.cc
+++ b/chrome/browser/ui/browser_ui_prefs.cc
@@ -227,6 +227,7 @@ void RegisterBrowserUserPrefs(user_prefs
     registry->RegisterBooleanPref(prefs::kHeliumServicesEnabled, true);
     registry->RegisterBooleanPref(prefs::kHeliumBangsEnabled, true);
     registry->RegisterBooleanPref(prefs::kHeliumExtProxyEnabled, true);
+    registry->RegisterBooleanPref(prefs::kHeliumSpellcheckEnabled, true);
     registry->RegisterStringPref(prefs::kHeliumServicesOrigin, "");
     registry->RegisterBooleanPref(prefs::kHeliumDidOnboarding, false);
     registry->RegisterBooleanPref(prefs::kHeliumServicesConsented, false);
--- a/chrome/browser/resources/settings/privacy_page/services_page.html
+++ b/chrome/browser/resources/settings/privacy_page/services_page.html
@@ -92,6 +92,13 @@
                 label="$i18n{heliumBangsToggle}"
                 sub-label="$i18n{heliumBangsToggleDescription}">
             </settings-toggle-button>
+            <if expr="use_renderer_spellchecker">
+                <settings-toggle-button id="spellcheckToggleButton"
+                    pref="{{prefs.helium.services.spellcheck_files}}"
+                    label="$i18n{heliumSpellcheckToggle}"
+                    sub-label="$i18n{heliumSpellcheckToggleDescription}">
+                </settings-toggle-button>
+            </if>
         </cr-collapse>
     </div>
     </template>
--- a/chrome/app/settings_strings.grdp
+++ b/chrome/app/settings_strings.grdp
@@ -1294,6 +1294,9 @@
         <message name="IDS_SETTINGS_LANGUAGES_DICTIONARY_DOWNLOAD_FAILED_HELP" desc="Error message when spell dictionary download fails more than once possibly due to a network policy or configuration.">
           Please check with your network administrator to make sure that the firewall is not blocking downloads from Google servers.
         </message>
+        <message name="IDS_SETTINGS_LANGUAGES_DICTIONARY_DOWNLOAD_FAILED_HELP_HELIUM" desc="Error message when spell dictionary download fails more than once possibly due to a network policy or configuration.">
+          Please check that Helium services are enabled (including spell check downloads) and make sure that the firewall is not blocking downloads from Helium servers.
+        </message>
       </if>
     </if>
     <if expr="is_chromeos">
@@ -1751,6 +1754,12 @@
     <message name="IDS_SETTINGS_HELIUM_SERVICES_BANGS_TOGGLE_DESCRIPTION" desc="Description of the toggle for enabling/disabling of downloading bangs">
       Helium will fetch a list of bangs that help you browse the Internet faster, such as !w or !gh. When disabled, bangs will not work.
     </message>
+    <message name="IDS_SETTINGS_HELIUM_SERVICES_SPELLCHECK_TOGGLE" desc="Toggle for enabling/disabling of downloading spellcheck files">
+      Allow downloading dictionary files for spell checking
+    </message>
+    <message name="IDS_SETTINGS_HELIUM_SERVICES_SPELLCHECK_TOGGLE_DESCRIPTION" desc="Description of the for enabling/disabling of downloading spellcheck files">
+      Helium will fetch dictionary files used for spell checking when requested. When disabled, spell checking will not work.
+    </message>
     <message name="IDS_SETTINGS_HELIUM_SERVICES_OVERRIDE" desc="Text input for overriding the Helium services server">
       Use your own instance of Helium services
     </message>
--- a/chrome/browser/ui/webui/settings/settings_localized_strings_provider.cc
+++ b/chrome/browser/ui/webui/settings/settings_localized_strings_provider.cc
@@ -1291,7 +1291,7 @@ void AddLanguagesStrings(content::WebUID
       {"languagesDictionaryDownloadError",
        IDS_SETTINGS_LANGUAGES_DICTIONARY_DOWNLOAD_FAILED},
       {"languagesDictionaryDownloadErrorHelp",
-       IDS_SETTINGS_LANGUAGES_DICTIONARY_DOWNLOAD_FAILED_HELP},
+       IDS_SETTINGS_LANGUAGES_DICTIONARY_DOWNLOAD_FAILED_HELP_HELIUM},
 #endif
 #endif  // !BUILDFLAG(IS_CHROMEOS)
 #if BUILDFLAG(IS_CHROMEOS)
@@ -2254,6 +2254,9 @@ void AddPrivacyStrings(content::WebUIDat
       {"heliumBangsToggle", IDS_SETTINGS_HELIUM_SERVICES_BANGS_TOGGLE},
       {"heliumBangsToggleDescription",
         IDS_SETTINGS_HELIUM_SERVICES_BANGS_TOGGLE_DESCRIPTION},
+      {"heliumSpellcheckToggle", IDS_SETTINGS_HELIUM_SERVICES_SPELLCHECK_TOGGLE},
+      {"heliumSpellcheckToggleDescription",
+        IDS_SETTINGS_HELIUM_SERVICES_SPELLCHECK_TOGGLE_DESCRIPTION},
       {"heliumOriginOverride", IDS_SETTINGS_HELIUM_SERVICES_OVERRIDE},
       {"heliumOriginOverrideDescription",
         IDS_SETTINGS_HELIUM_SERVICES_OVERRIDE_DESCRIPTION},
--- a/chrome/common/features.gni
+++ b/chrome/common/features.gni
@@ -9,6 +9,7 @@ import("//build/config/features.gni")
 import("//components/compose/features.gni")
 import("//components/safe_browsing/buildflags.gni")
 import("//components/signin/features.gni")
+import("//components/spellcheck/spellcheck_build_features.gni")
 import("//components/supervised_user/buildflags.gni")
 import("//content/public/common/features.gni")
 import("//crypto/features.gni")
@@ -110,4 +111,5 @@ chrome_grit_defines = [
   "chrome_root_store_cert_management_ui=$chrome_root_store_cert_management_ui",
   "use_nss_certs=$use_nss_certs",
   "webnn_enable_graph_dump=$webnn_enable_graph_dump",
+  "use_renderer_spellchecker=$use_renderer_spellchecker",
 ]
