--- a/chrome/app/settings_strings.grdp
+++ b/chrome/app/settings_strings.grdp
@@ -1705,6 +1705,50 @@
       </message>
     </if>
 
+    <!-- Nitrous Services Privacy Settings (nitrous://settings/privacy/services) -->
+    <message name="IDS_SETTINGS_HELIUM_SERVICES" desc="Name of the page which controls connectivity to Nitrous services">
+      Nitrous services
+    </message>
+    <message name="IDS_SETTINGS_HELIUM_SERVICES_DESCRIPTION" desc="Description of the controls available on the Nitrous services settings page">
+      Manage what Nitrous services are allowed in your browser
+    </message>
+    <message name="IDS_SETTINGS_HELIUM_SCHEMA_NOTICE_TITLE" desc="Title of the alert shown when Nitrous services have a change in behavior">
+      Updates to Nitrous services:
+    </message>
+    <message name="IDS_SETTINGS_HELIUM_SCHEMA_NOTICE_DESCRIPTION" desc="Description of the alert shown when Nitrous services have a change in behavior">
+      These changes are not active until you dismiss this notice.
+    </message>
+    <message name="IDS_SETTINGS_HELIUM_SCHEMA_IGNORE" desc="Checkbox label to ignore any future services change alerts">
+      Don't notify me again
+    </message>
+    <message name="IDS_SETTINGS_HELIUM_SERVICES_TOGGLE" desc="Global toggle for enabling/disabling of all Nitrous services">
+      Allow connecting to Nitrous services
+    </message>
+    <message name="IDS_SETTINGS_HELIUM_SERVICES_TOGGLE_DESCRIPTION" desc="Description of the toggle for enabling/disabling of all Nitrous services">
+      When enabled, Nitrous will be able to connect to anonymous web services to provide additional functionality. When disabled, additional features will not work.
+    </message>
+    <message name="IDS_SETTINGS_HELIUM_SERVICES_OVERRIDE" desc="Text input for overriding the Nitrous services server">
+      Use your own instance of Nitrous services
+    </message>
+    <message name="IDS_SETTINGS_HELIUM_SERVICES_OVERRIDE_DESCRIPTION" desc="Description of the text input for overriding the Nitrous services server">
+      You can host your own instance of Nitrous services and use it in your browser instead of the default one. HTTPS only.
+    </message>
+    <message name="IDS_SETTINGS_HELIUM_SERVICES_OVERRIDE_DESCRIPTION_WARNING" desc="Warning section of description for overriding the Nitrous services server">
+      Do not use this field unless you know exactly what you are doing. Never paste URLs from other people here, they are trying to steal your data. We are not responsible for any damages caused by using a custom server and will not be able to help you.
+    </message>
+    <message name="IDS_SETTINGS_HELIUM_SERVICES_OVERRIDE_ARIA_LABEL" desc="ARIA label of the text input for overriding the Nitrous services server">
+      Custom origin URL for Nitrous services
+    </message>
+    <message name="IDS_SETTINGS_HELIUM_SERVICES_OVERRIDE_INVALID_ORIGIN" desc="Validation error shown when the custom Nitrous services origin is invalid.">
+      Invalid origin
+    </message>
+    <message name="IDS_SETTINGS_PERSONA_TITLE" desc="Name of the page used to manage browser persona settings">
+      Persona
+    </message>
+    <message name="IDS_SETTINGS_PERSONA_DESCRIPTION" desc="Description of the browser persona settings page">
+      Configure browser persona and fingerprint surface overrides
+    </message>
+
     <!-- Safety check -->
     <message name="IDS_SETTINGS_SAFETY_CHECK_ICON_INFO_ARIA_LABEL" desc="Accessibility text for a safety check child icon, that indicates info for the child check.">
       Info
--- a/chrome/browser/ui/webui/settings/settings_localized_strings_provider.cc
+++ b/chrome/browser/ui/webui/settings/settings_localized_strings_provider.cc
@@ -2234,6 +2234,27 @@ void AddPrivacyStrings(content::WebUIDat
       {"siteSettingsSublabel", IDS_SETTINGS_PERMISSIONS_DESCRIPTION},
       {"securityPageTitle", IDS_SETTINGS_SECURITY},
       {"securityPageDescription", IDS_SETTINGS_SECURITY_DESCRIPTION},
+      {"heliumServicesTitle", IDS_SETTINGS_HELIUM_SERVICES},
+      {"heliumServicesDescription", IDS_SETTINGS_HELIUM_SERVICES_DESCRIPTION},
+      {"heliumServicesToggle", IDS_SETTINGS_HELIUM_SERVICES_TOGGLE},
+      {"heliumServicesToggleDescription",
+        IDS_SETTINGS_HELIUM_SERVICES_TOGGLE_DESCRIPTION},
+      {"heliumSchemaUpdatedTitle", IDS_SETTINGS_HELIUM_SCHEMA_NOTICE_TITLE},
+      {"heliumSchemaUpdatedDescription",
+        IDS_SETTINGS_HELIUM_SCHEMA_NOTICE_DESCRIPTION},
+      {"heliumGotIt", IDS_SETTINGS_GOT_IT},
+      {"heliumSchemaIgnore", IDS_SETTINGS_HELIUM_SCHEMA_IGNORE},
+      {"heliumOriginOverride", IDS_SETTINGS_HELIUM_SERVICES_OVERRIDE},
+      {"heliumOriginOverrideDescription",
+        IDS_SETTINGS_HELIUM_SERVICES_OVERRIDE_DESCRIPTION},
+      {"heliumOriginOverrideDescriptionWarning",
+        IDS_SETTINGS_HELIUM_SERVICES_OVERRIDE_DESCRIPTION_WARNING},
+      {"heliumOriginOverrideLabel",
+        IDS_SETTINGS_HELIUM_SERVICES_OVERRIDE_ARIA_LABEL},
+      {"heliumOriginOverrideInvalidOrigin",
+        IDS_SETTINGS_HELIUM_SERVICES_OVERRIDE_INVALID_ORIGIN},
+      {"personaTitle", IDS_SETTINGS_PERSONA_TITLE},
+      {"personaDescription", IDS_SETTINGS_PERSONA_DESCRIPTION},
       {"advancedProtectionProgramTitle",
        IDS_SETTINGS_ADVANCED_PROTECTION_PROGRAM},
       {"advancedProtectionProgramDesc",
--- a/chrome/browser/resources/settings/privacy_page/privacy_page.html
+++ b/chrome/browser/resources/settings/privacy_page/privacy_page.html
@@ -46,6 +46,11 @@
         sub-label="$i18n{siteSettingsSublabel}"
         on-click="onSiteSettingsLinkRowClick_"
         role-description="$i18n{subpageArrowRoleDescription}"></cr-link-row>
+    <cr-link-row id="heliumServicesLinkRow" start-icon="cr:chrome-product"
+        class="hr" label="$i18n{heliumServicesTitle}"
+        sub-label="$i18n{heliumServicesDescription}"
+        on-click="onHeliumServicesPageClick_"
+        role-description="$i18n{subpageArrowRoleDescription}"></cr-link-row>
     <cr-toast id="deleteBrowsingDataToast" duration="5000">
       <div>[[dbdDeletionConfirmationToastLabel_]]</div>
     </cr-toast>
--- a/chrome/browser/resources/settings/privacy_page/privacy_page.ts
+++ b/chrome/browser/resources/settings/privacy_page/privacy_page.ts
@@ -165,6 +165,11 @@
     Router.getInstance().navigateTo(routes.SITE_SETTINGS);
   }
 
+  private onHeliumServicesPageClick_() {
+    this.interactedWithPage_();
+    Router.getInstance().navigateTo(routes.HELIUM_SERVICES);
+  }
+
   private onSecurityPageClick_() {
     this.interactedWithPage_();
     this.metricsBrowserProxy_.recordAction(
--- a/chrome/browser/resources/settings/privacy_page/privacy_page_index.html
+++ b/chrome/browser/resources/settings/privacy_page/privacy_page_index.html
@@ -39,6 +39,14 @@
     </settings-privacy-page>
   </template>
 
+  <template is="dom-if" if="[[renderView_(
+      routes_.HELIUM_SERVICES, currentRoute, inSearchMode)]]">
+    <settings-helium-services-page slot="view" id="heliumServices"
+        data-parent-view-id="privacy" prefs="{{prefs}}"
+        route-path$="[[routes_.HELIUM_SERVICES.path]]">
+    </settings-helium-services-page>
+  </template>
+
   <template is="dom-if" if="[[isAdPrivacyAvailable_]]" restamp>
     <template is="dom-if" if="[[renderView_(
         routes_.PRIVACY_SANDBOX, currentRoute, inSearchMode)]]"
--- a/chrome/browser/resources/settings/router.ts
+++ b/chrome/browser/resources/settings/router.ts
@@ -34,6 +34,7 @@ export interface SettingsRoutes {
   GEMINI: Route;
   GEMINI_LOGIN: Route;
   GLIC_SECTION: Route;
+  HELIUM_SERVICES: Route;
   HISTORY_SEARCH: Route;
   LANGUAGES: Route;
   MANAGE_PROFILE: Route;
--- a/chrome/browser/resources/settings/route.ts
+++ b/chrome/browser/resources/settings/route.ts
@@ -18,6 +18,8 @@ function addPrivacyChildRoutes(r: Partia
   r.CLEAR_BROWSER_DATA = r.PRIVACY.createChild('/clearBrowserData');
   r.CLEAR_BROWSER_DATA.isNavigableDialog = true;
 
+  r.HELIUM_SERVICES = r.PRIVACY.createChild('services');
+
   const visibility = pageVisibility || {};
 
   if (visibility.safetyHub !== false) {
--- a/chrome/browser/resources/settings/BUILD.gn
+++ b/chrome/browser/resources/settings/BUILD.gn
@@ -157,6 +157,7 @@ build_webui("build") {
     "privacy_page/security/security_page.ts",
     "privacy_page/security/security_page_feature_row.ts",
     "privacy_page/security/security_page_v2.ts",
+    "privacy_page/services_page.ts",
     "privacy_sandbox/privacy_sandbox_ad_measurement_subpage.ts",
     "privacy_sandbox/privacy_sandbox_fledge_subpage.ts",
     "privacy_sandbox/privacy_sandbox_interest_item.ts",
@@ -395,6 +396,7 @@ build_webui("build") {
       "languages_page/languages_util.ts",
       "people_page/import_data_browser_proxy.ts",
       "people_page/manage_profile_browser_proxy.ts",
+      "privacy_page/changelog_proxy.ts",
       "system_page/system_page_browser_proxy.ts",
     ]
     if (is_chrome_branded) {
--- /dev/null
+++ b/chrome/browser/resources/settings/privacy_page/services_page.html
@@ -0,0 +1,89 @@
+<style include="cr-shared-style settings-shared">
+    .label-wrapper {
+        padding: var(--cr-section-vertical-padding) 0;
+    }
+
+    #changelog-notice {
+        --cr-changelog-padding: 16px;
+        border-radius: calc(var(--cr-changelog-padding) / 2);
+        margin: var(--cr-changelog-padding);
+        margin-top: 0;
+        padding: var(--cr-changelog-padding);
+        background: var(--md-background-color);
+    }
+
+    #changelog-title {
+        font-size: 15px;
+        font-weight: 500;
+    }
+
+    #changelog-notice-actions {
+        --cr-checkbox-label-padding-start: 8px;
+        display: flex;
+        gap: var(--cr-changelog-padding);
+        border-top: var(--cr-separator-line);
+        margin-top: var(--cr-changelog-padding);
+        padding-top: calc(var(--cr-changelog-padding) / 2);
+    }
+</style>
+
+<settings-subpage page-title="$i18n{heliumServicesTitle}" route-path$="[[routePath]]">
+<div class="cr-col first">
+        <template is="dom-if" if="[[changes_.length]]">
+          <div id="changelog-notice">
+            <div id="changelog-title">$i18n{heliumSchemaUpdatedTitle}</div>
+            <ul>
+                <template is="dom-repeat" items="[[changes_]]" as="change">
+                    <li>[[change]]</li>
+                </template>
+            </ul>
+            <div id="changelog-description">
+                $i18n{heliumSchemaUpdatedDescription}
+            </div>
+            <div id="changelog-notice-actions">
+                <cr-button on-click="onAcknowledgedChangelog_">$i18n{heliumGotIt}</cr-button>
+                <cr-checkbox id="checkbox" checked="{{ignoreChecked_}}">
+                    $i18n{heliumSchemaIgnore}
+                </cr-checkbox>
+            </div>
+          </div>
+        </template>
+    <settings-toggle-button id="servicesToggleButton"
+        pref="{{prefs.nitrous.services.enabled}}"
+        label="$i18n{heliumServicesToggle}"
+        sub-label="$i18n{heliumServicesToggleDescription}">
+    </settings-toggle-button>
+    <div class="cr-col">
+        <cr-collapse id="servicesCollapse" opened="[[prefs.nitrous.services.enabled.value]]">
+        </cr-collapse>
+    </div>
+</div>
+
+
+<div class="cr-row">
+  <div class="flex">
+    <div class="flex label-wrapper">
+        <div class="label">$i18n{heliumOriginOverride}</div>
+        <div class="cr-secondary-text label">
+            $i18n{heliumOriginOverrideDescription}
+            <br><br>
+            <b>$i18n{heliumOriginOverrideDescriptionWarning}</b>
+        </div>
+    </div>
+    <cr-input
+        id="heliumOriginOverride"
+        placeholder="https://services.helium.imput.net"
+        value="[[prefs.nitrous.services.origin_override.value]]"
+        pattern="https://.*"
+        on-keydown="onOriginOverrideKeydown_"
+        on-change="onOriginOverrideChange_"
+        type="url"
+        maxlength="128"
+        aria-label="$i18n{heliumOriginOverrideLabel}"
+        auto-validate
+        error-message="$i18n{heliumOriginOverrideInvalidOrigin}"
+        spellcheck="false">
+    </cr-input>
+  </div>
+</div>
+</settings-subpage>
--- /dev/null
+++ b/chrome/browser/resources/settings/privacy_page/services_page.ts
@@ -0,0 +1,105 @@
+// Copyright 2025 The Helium Authors
+// You can use, redistribute, and/or modify this source code under
+// the terms of the GPL-3.0 license that can be found in the LICENSE file.
+
+import '../controls/settings_toggle_button.js';
+import '../settings_page/settings_subpage.js';
+import 'chrome://resources/cr_elements/cr_button/cr_button.js';
+import 'chrome://resources/cr_elements/cr_checkbox/cr_checkbox.js';
+import 'chrome://resources/cr_elements/cr_collapse/cr_collapse.js';
+import 'chrome://resources/cr_elements/cr_icon/cr_icon.js';
+import 'chrome://resources/cr_elements/cr_input/cr_input.js';
+
+import {PrefsMixin} from '/shared/settings/prefs/prefs_mixin.js';
+import {HelpBubbleMixin} from 'chrome://resources/cr_components/help_bubble/help_bubble_mixin.js';
+import type {CrInputElement} from 'chrome://resources/cr_elements/cr_input/cr_input.js';
+import {I18nMixin} from 'chrome://resources/cr_elements/i18n_mixin.js';
+import {PolymerElement} from 'chrome://resources/polymer/v3_0/polymer/polymer_bundled.min.js';
+
+import type {HeliumServicesChangeHandler} from './changelog_proxy.js';
+import {HeliumServicesChangeHandlerImpl} from './changelog_proxy.js';
+import {getTemplate} from './services_page.html.js';
+
+const SettingsHeliumServicesPageElementBase =
+    HelpBubbleMixin(I18nMixin(PrefsMixin(PolymerElement)));
+
+export class SettingsHeliumServicesPageElement extends
+    SettingsHeliumServicesPageElementBase {
+  private changeProxy_: HeliumServicesChangeHandler =
+      HeliumServicesChangeHandlerImpl.getInstance();
+
+  static get is() {
+    return 'settings-helium-services-page';
+  }
+
+  static get template() {
+    return getTemplate();
+  }
+
+  static get properties() {
+    return {
+      ignoreChecked_: Boolean,
+      changes_: Array,
+    };
+  }
+
+  declare private changes_: string[];
+  declare private ignoreChecked_: boolean;
+
+  private toOrigin(thing: string) {
+    try {
+      return new URL(thing).origin;
+    } catch {
+      return null;
+    }
+  }
+
+  private updateChangelogState(changes: string[]) {
+    this.changes_ = changes.flatMap(ch => ch.split('\n'));
+  }
+
+  private requestChangelog() {
+    this.changeProxy_.requestChangelog().then(
+        this.updateChangelogState.bind(this));
+  }
+
+  private onAcknowledgedChangelog_() {
+    this.changeProxy_.acknowledgeChanges(this.ignoreChecked_);
+    this.requestChangelog();
+  }
+
+  override ready() {
+    super.ready();
+    this.requestChangelog();
+  }
+
+  protected onOriginOverrideKeydown_(event: KeyboardEvent) {
+    const target = event.target as CrInputElement;
+
+    if (event.key === 'Escape' || event.key === 'Enter') {
+      target.blur();
+    }
+  }
+
+  private onOriginOverrideChange_(event: Event) {
+    const target = event.target as CrInputElement;
+    if (target.invalid) {
+      return;
+    }
+
+    const origin = this.toOrigin(target.value);
+    if (origin) {
+      target.value = origin;
+    }
+
+    this.setPrefValue('nitrous.services.origin_override', target.value);
+  }
+}
+
+declare global {
+    interface HTMLElementTagNameMap {
+      'settings-helium-services-page': SettingsHeliumServicesPageElement;
+    }
+}
+
+customElements.define(SettingsHeliumServicesPageElement.is, SettingsHeliumServicesPageElement);
--- a/chrome/browser/resources/settings/lazy_load.ts
+++ b/chrome/browser/resources/settings/lazy_load.ts
@@ -27,6 +27,7 @@ import './privacy_page/privacy_guide/pri
 import './privacy_page/security/security_keys_subpage.js';
 import './privacy_page/security/security_page_v2.js';
 import './privacy_page/security/security_page.js';
+import './privacy_page/services_page.js';
 import './privacy_sandbox/privacy_sandbox_ad_measurement_subpage.js';
 import './privacy_sandbox/privacy_sandbox_fledge_subpage.js';
 import './privacy_sandbox/privacy_sandbox_interest_item.js';
@@ -271,6 +272,7 @@ export {SecurityKeysSubpageElement} from
 export {HttpsFirstModeSetting, SettingsSecurityPageElement} from './privacy_page/security/security_page.js';
 export {SecurityPageFeatureRowElement} from './privacy_page/security/security_page_feature_row.js';
 export {SecuritySettingsBundleSetting, SettingsSecurityPageV2Element} from './privacy_page/security/security_page_v2.js';
+export {SettingsHeliumServicesPageElement} from './privacy_page/services_page.js';
 export {SettingsPrivacySandboxAdMeasurementSubpageElement} from './privacy_sandbox/privacy_sandbox_ad_measurement_subpage.js';
 export {SettingsPrivacySandboxFledgeSubpageElement} from './privacy_sandbox/privacy_sandbox_fledge_subpage.js';
 export {PrivacySandboxInterestItemElement} from './privacy_sandbox/privacy_sandbox_interest_item.js';
--- a/chrome/browser/ui/browser_ui_prefs.cc
+++ b/chrome/browser/ui/browser_ui_prefs.cc
@@ -15,6 +15,7 @@
 #include "chrome/browser/upgrade_detector/upgrade_detector.h"
 #include "chrome/common/buildflags.h"
 #include "chrome/common/pref_names.h"
+#include "components/helium_services/pref_names.h"
 #include "components/policy/core/common/policy_pref_names.h"
 #include "components/pref_registry/pref_registry_syncable.h"
 #include "components/prefs/pref_registry_simple.h"
@@ -215,6 +216,13 @@ void RegisterBrowserUserPrefs(user_prefs
                                 false);
 #endif
 
+  {
+    registry->RegisterIntegerPref(prefs::kHeliumSchemaVersion, 0);
+    registry->RegisterBooleanPref(prefs::kHeliumDisableSchemaAlerts, false);
+    registry->RegisterBooleanPref(prefs::kHeliumServicesEnabled, true);
+    registry->RegisterStringPref(prefs::kHeliumServicesOrigin, "");
+  }
+
   registry->RegisterBooleanPref(
       prefs::kHttpsOnlyModeEnabled, false,
       user_prefs::PrefRegistrySyncable::SYNCABLE_PREF);
--- a/chrome/browser/extensions/api/settings_private/prefs_util.cc
+++ b/chrome/browser/extensions/api/settings_private/prefs_util.cc
@@ -46,6 +46,7 @@
 #include "components/dom_distiller/core/pref_names.h"
 #include "components/drive/drive_pref_names.h"
 #include "components/embedder_support/pref_names.h"
+#include "components/helium_services/pref_names.h"
 #include "components/language/core/browser/pref_names.h"
 #include "components/live_caption/pref_names.h"
 #include "components/media_router/common/pref_names.h"
@@ -396,6 +397,12 @@
       settings_api::PrefType::kString;
 #endif
 
+  // Helium services page
+  (*s_allowlist)[::prefs::kHeliumServicesEnabled] =
+      settings_api::PrefType::kBoolean;
+  (*s_allowlist)[::prefs::kHeliumServicesOrigin] =
+      settings_api::PrefType::kString;
+
   // Privacy Guide
   (*s_allowlist)[::prefs::kPrivacyGuideViewed] =
       settings_api::PrefType::kBoolean;
--- /dev/null
+++ b/components/helium_services/BUILD.gn
@@ -0,0 +1,22 @@
+# Copyright 2025 The Helium Authors
+# You can use, redistribute, and/or modify this source code under
+# the terms of the GPL-3.0 license that can be found in the LICENSE file.
+
+component("helium_services") {
+  sources = [
+    "helium_services_helpers.cc",
+    "helium_services_helpers.h",
+    "pref_names.h",
+    "schema.cc",
+    "schema.h",
+  ]
+
+  public_deps = [
+    "//base",
+    "//components/prefs",
+    "//url",
+    "//net",
+  ]
+
+  defines = [ "IS_HELIUM_IMPL" ]
+}
--- /dev/null
+++ b/components/helium_services/helium_services_helpers.cc
@@ -0,0 +1,67 @@
+// Copyright 2025 The Helium Authors
+// You can use, redistribute, and/or modify this source code under
+// the terms of the GPL-3.0 license that can be found in the LICENSE file.
+
+#include "components/helium_services/helium_services_helpers.h"
+
+#include <optional>
+
+#include "base/functional/bind.h"
+#include "components/helium_services/pref_names.h"
+#include "components/prefs/pref_service.h"
+#include "net/base/url_util.h"
+#include "url/gurl.h"
+#include "url/url_constants.h"
+
+namespace helium {
+
+std::optional<GURL> GetValidUserOverridenURL(std::string_view user_url_) {
+    if (user_url_.empty()) {
+        return std::nullopt;
+    }
+
+    GURL user_url = GURL(user_url_);
+    if (!user_url.is_valid()) {
+        return std::nullopt;
+    }
+
+    bool isSecure = user_url.SchemeIs(url::kHttpsScheme) || net::IsLocalhost(user_url);
+    if (!isSecure) {
+        return std::nullopt;
+    }
+
+    return user_url;
+}
+
+std::optional<GURL> GetValidUserOverridenURL(const PrefService& prefs) {
+    return GetValidUserOverridenURL(
+        prefs.GetString(prefs::kHeliumServicesOrigin)
+    );
+}
+
+GURL GetServicesBaseURL(const PrefService& prefs) {
+    std::optional<GURL> user_url = GetValidUserOverridenURL(prefs);
+    if (user_url) {
+        return *user_url;
+    }
+
+    return GURL(kHeliumDefaultOrigin);
+}
+
+GURL GetDummyURL() {
+    return GURL(kHeliumDummyOrigin);
+}
+
+bool ShouldAccessServices(const PrefService& prefs) {
+    return prefs.GetBoolean(prefs::kHeliumServicesEnabled);
+}
+
+void ConfigurePrefChangeRegistrarFor(std::string_view pref_name,
+    PrefChangeRegistrar& registrar, const base::RepeatingClosure& observer) {
+  registrar.Add(prefs::kHeliumServicesEnabled, observer);
+  registrar.Add(prefs::kHeliumServicesOrigin, observer);
+  registrar.Add(prefs::kHeliumSchemaVersion, observer);
+  registrar.Add(pref_name, observer);
+}
+
+}  // namespace helium
--- /dev/null
+++ b/components/helium_services/helium_services_helpers.h
@@ -0,0 +1,30 @@
+// Copyright 2025 The Helium Authors
+// You can use, redistribute, and/or modify this source code under
+// the terms of the GPL-3.0 license that can be found in the LICENSE file.
+
+#ifndef COMPONENTS_HELIUM_SERVICES_HELIUM_SERVICES_HELPERS_H_
+#define COMPONENTS_HELIUM_SERVICES_HELIUM_SERVICES_HELPERS_H_
+
+#include "base/functional/bind.h"
+#include "components/prefs/pref_change_registrar.h"
+#include "components/prefs/pref_service.h"
+#include "url/gurl.h"
+
+namespace helium {
+
+const char kHeliumDefaultOrigin[] =
+    "https://services.helium.imput.net";
+
+const char kHeliumDummyOrigin[] =
+    "https://helium-services-are-disabled.qjz9zk";
+
+bool ShouldAccessServices(const PrefService& prefs);
+
+COMPONENT_EXPORT(HELIUM) GURL GetServicesBaseURL(const PrefService& prefs);
+COMPONENT_EXPORT(HELIUM) GURL GetDummyURL();
+COMPONENT_EXPORT(HELIUM) std::optional<GURL> GetValidUserOverridenURL(std::string_view user_url_);
+COMPONENT_EXPORT(HELIUM) void ConfigurePrefChangeRegistrarFor(std::string_view pref_name,
+                            PrefChangeRegistrar& registrar, const base::RepeatingClosure& observer);
+}  // namespace helium
+
+#endif  // COMPONENTS_HELIUM_SERVICES_HELIUM_SERVICES_HELPERS_H_
--- /dev/null
+++ b/components/helium_services/pref_names.h
@@ -0,0 +1,24 @@
+// Copyright 2025 The Helium Authors
+// You can use, redistribute, and/or modify this source code under
+// the terms of the GPL-3.0 license that can be found in the LICENSE file.
+
+#ifndef COMPONENTS_HELIUM_SERVICES_PREF_NAMES_H_
+#define COMPONENTS_HELIUM_SERVICES_PREF_NAMES_H_
+
+namespace prefs {
+
+inline constexpr char kHeliumSchemaVersion[] =
+    "nitrous.services.schema_version";
+
+inline constexpr char kHeliumDisableSchemaAlerts[] =
+    "nitrous.services.disable_schema_alerts";
+
+inline constexpr char kHeliumServicesEnabled[] =
+    "nitrous.services.enabled";
+
+inline constexpr char kHeliumServicesOrigin[] =
+    "nitrous.services.origin_override";
+
+}  // namespace prefs
+
+#endif  // COMPONENTS_HELIUM_SERVICES_PREF_NAMES_H_
--- /dev/null
+++ b/chrome/browser/resources/settings/privacy_page/changelog_proxy.ts
@@ -0,0 +1,40 @@
+// Copyright 2025 The Helium Authors
+// You can use, redistribute, and/or modify this source code under
+// the terms of the GPL-3.0 license that can be found in the LICENSE file.
+
+/**
+ * @fileoverview A helper object used from the "Helium services" section
+ * to request a changelog (if one exists).
+ */
+
+// clang-format off
+import {sendWithPromise} from 'chrome://resources/js/cr.js';
+// clang-format on
+
+export interface HeliumServicesChangeHandler {
+  // Get the pending changes to Helium services to announce
+  // to the user.
+  requestChangelog(): Promise<string[]>;
+  acknowledgeChanges(ignoreAllChangelogs: boolean): void;
+}
+
+export class HeliumServicesChangeHandlerImpl implements
+    HeliumServicesChangeHandler {
+  requestChangelog() {
+    return sendWithPromise<string[]>('getServicesSchemaChangelog');
+  }
+
+  acknowledgeChanges(ignoreAllChangelogs: boolean) {
+    chrome.send('updateServicesSchemaVersion', [ignoreAllChangelogs]);
+  }
+
+  static getInstance(): HeliumServicesChangeHandler {
+    return instance || (instance = new HeliumServicesChangeHandlerImpl());
+  }
+
+  static setInstance(obj: HeliumServicesChangeHandler) {
+    instance = obj;
+  }
+}
+
+let instance: HeliumServicesChangeHandler|null = null;
--- /dev/null
+++ b/chrome/browser/ui/webui/settings/services_schema_handler.h
@@ -0,0 +1,36 @@
+// Copyright 2025 The Helium Authors
+// You can use, redistribute, and/or modify this source code under
+// the terms of the GPL-3.0 license that can be found in the LICENSE file.
+
+#ifndef CHROME_BROWSER_UI_WEBUI_SETTINGS_SERVICES_SCHEMA_HANDLER_H_
+#define CHROME_BROWSER_UI_WEBUI_SETTINGS_SERVICES_SCHEMA_HANDLER_H_
+
+#include <optional>
+#include <string>
+
+#include "base/memory/raw_ptr.h"
+#include "base/values.h"
+#include "content/public/browser/web_ui_message_handler.h"
+
+class Profile;
+class PrefService;
+
+class HeliumServicesSchemaHandler : public content::WebUIMessageHandler {
+ public:
+  explicit HeliumServicesSchemaHandler(Profile* profile);
+  HeliumServicesSchemaHandler(const HeliumServicesSchemaHandler&) = delete;
+  HeliumServicesSchemaHandler& operator=(const HeliumServicesSchemaHandler&) =
+      delete;
+  ~HeliumServicesSchemaHandler() override;
+
+  // SettingsPageUIHandler:
+  void RegisterMessages() override;
+
+ private:
+  void HandleChangelogRequest(const base::ListValue&);
+  void HandleChangelogAcknowledgment(const base::ListValue&);
+
+  raw_ptr<PrefService> pref_service_ = nullptr;
+};
+
+#endif  // CHROME_BROWSER_UI_WEBUI_SETTINGS_SERVICES_SCHEMA_HANDLER_H_
--- /dev/null
+++ b/components/helium_services/schema.cc
@@ -0,0 +1,40 @@
+// Copyright 2025 The Helium Authors
+// You can use, redistribute, and/or modify this source code under
+// the terms of the GPL-3.0 license that can be found in the LICENSE file.
+
+#include "components/helium_services/schema.h"
+
+#include "components/helium_services/helium_services_helpers.h"
+#include "components/helium_services/pref_names.h"
+#include "components/prefs/pref_service.h"
+
+namespace helium {
+
+bool HasAcceptedSchema(const PrefService& prefs, int version) {
+  return prefs.GetBoolean(prefs::kHeliumDisableSchemaAlerts) ||
+         prefs.GetInteger(prefs::kHeliumSchemaVersion) >= version;
+}
+
+void AcceptCurrentSchema(PrefService& prefs) {
+  prefs.SetInteger(prefs::kHeliumSchemaVersion, kHeliumCurrentSchemaVersion);
+}
+
+bool ShouldShowSchemaNotification(const PrefService& prefs) {
+  return ShouldAccessServices(prefs) &&
+         !HasAcceptedSchema(prefs, kHeliumCurrentSchemaVersion);
+}
+
+ServicesChangelog& GetChangelog() {
+  static constexpr auto kHeliumSchemaChangelog =
+      base::MakeFixedFlatMap<int, std::string_view>({
+          {
+            1,
+            "Automatic component updates are now available. They're managed by the same toggle that enables automatic browser updates.\n"
+            "From now on, you'll be notified about major changes to Nitrous services. Even though these notifications are extremely rare, you can choose to ignore them and accept all future changes automatically."
+          }
+      });
+
+  return kHeliumSchemaChangelog;
+}
+
+}  // namespace helium
--- /dev/null
+++ b/components/helium_services/schema.h
@@ -0,0 +1,30 @@
+// Copyright 2025 The Helium Authors
+// You can use, redistribute, and/or modify this source code under
+// the terms of the GPL-3.0 license that can be found in the LICENSE file.
+
+#ifndef COMPONENTS_HELIUM_SERVICES_SCHEMA_H_
+#define COMPONENTS_HELIUM_SERVICES_SCHEMA_H_
+
+#include "base/component_export.h"
+#include "base/containers/fixed_flat_map.h"
+#include "components/prefs/pref_service.h"
+
+namespace helium {
+
+inline constexpr int kHeliumCurrentSchemaVersion = 1;
+
+using ServicesChangelog = const base::
+    fixed_flat_map<int, std::string_view, kHeliumCurrentSchemaVersion>;
+
+#define EX COMPONENT_EXPORT(HELIUM)
+
+EX bool HasAcceptedSchema(const PrefService& prefs, int version);
+EX void AcceptCurrentSchema(PrefService& prefs);
+EX bool ShouldShowSchemaNotification(const PrefService& prefs);
+EX ServicesChangelog& GetChangelog();
+
+#undef EX
+
+}  // namespace helium
+
+#endif  // COMPONENTS_HELIUM_SERVICES_SCHEMA_H_
--- a/chrome/browser/ui/BUILD.gn
+++ b/chrome/browser/ui/BUILD.gn
@@ -535,6 +535,7 @@ static_library("ui") {
     "//components/download/content/factory",
     "//components/download/content/public",
     "//components/embedder_support",
+    "//components/helium_services",
     "//components/embedder_support:user_agent",
     "//components/encrypted_messages:encrypted_message_proto",
     "//components/enterprise",
@@ -1505,6 +1505,8 @@ static_library("ui") {
       "webui/settings/search_engines_handler.h",
       "webui/settings/security_settings_provider.cc",
       "webui/settings/security_settings_provider.h",
+      "webui/settings/services_schema_handler.cc",
+      "webui/settings/services_schema_handler.h",
       "webui/settings/settings_clear_browsing_data_handler.cc",
       "webui/settings/settings_clear_browsing_data_handler.h",
       "webui/settings/settings_localized_strings_privacy_sandbox_provider.cc",
--- /dev/null
+++ b/chrome/browser/ui/webui/settings/services_schema_handler.cc
@@ -0,0 +1,58 @@
+// Copyright 2025 The Helium Authors
+// You can use, redistribute, and/or modify this source code under
+// the terms of the GPL-3.0 license that can be found in the LICENSE file.
+
+#include "chrome/browser/ui/webui/settings/services_schema_handler.h"
+
+#include "base/functional/bind.h"
+#include "base/values.h"
+#include "chrome/browser/profiles/profile.h"
+#include "components/helium_services/pref_names.h"
+#include "components/helium_services/schema.h"
+#include "components/prefs/pref_service.h"
+
+HeliumServicesSchemaHandler::HeliumServicesSchemaHandler(Profile* profile)
+    : pref_service_(profile->GetPrefs()) {}
+
+HeliumServicesSchemaHandler::~HeliumServicesSchemaHandler() = default;
+
+void HeliumServicesSchemaHandler::RegisterMessages() {
+  web_ui()->RegisterMessageCallback(
+      "getServicesSchemaChangelog",
+      base::BindRepeating(&HeliumServicesSchemaHandler::HandleChangelogRequest,
+                          base::Unretained(this)));
+  web_ui()->RegisterMessageCallback(
+      "updateServicesSchemaVersion",
+      base::BindRepeating(
+          &HeliumServicesSchemaHandler::HandleChangelogAcknowledgment,
+          base::Unretained(this)));
+}
+
+void HeliumServicesSchemaHandler::HandleChangelogRequest(
+    const base::ListValue& args) {
+  CHECK_EQ(1U, args.size());
+  const auto& callback_id = args[0].GetString();
+  base::ListValue out;
+
+  if (helium::ShouldShowSchemaNotification(*pref_service_)) {
+    int user_version = pref_service_->GetInteger(prefs::kHeliumSchemaVersion);
+    auto& changelog = helium::GetChangelog();
+
+    for (auto it = changelog.upper_bound(user_version); it != changelog.end();
+         ++it) {
+      out.Append(it->second);
+    }
+  }
+
+  AllowJavascript();
+  ResolveJavascriptCallback(base::Value(callback_id), std::move(out));
+}
+
+void HeliumServicesSchemaHandler::HandleChangelogAcknowledgment(
+    const base::ListValue& args) {
+  CHECK_EQ(1U, args.size());
+  bool ignore_alerts = args[0].GetBool();
+
+  helium::AcceptCurrentSchema(*pref_service_);
+  pref_service_->SetBoolean(prefs::kHeliumDisableSchemaAlerts, ignore_alerts);
+}
--- a/chrome/browser/ui/webui/settings/settings_ui.cc
+++ b/chrome/browser/ui/webui/settings/settings_ui.cc
@@ -82,6 +82,7 @@
 #include "chrome/browser/ui/webui/settings/saved_info_handler.h"
 #include "chrome/browser/ui/webui/settings/search_engines_handler.h"
 #include "chrome/browser/ui/webui/settings/security_settings_provider.h"
+#include "chrome/browser/ui/webui/settings/services_schema_handler.h"
 #include "chrome/browser/ui/webui/settings/settings_clear_browsing_data_handler.h"
 #include "chrome/browser/ui/webui/settings/settings_localized_strings_provider.h"
 #include "chrome/browser/ui/webui/settings/settings_media_devices_selection_handler.h"
@@ -272,6 +273,8 @@ SettingsUI::SettingsUI(content::WebUI* w
   AddSettingsPageUIHandler(std::make_unique<ProtocolHandlersHandler>(profile));
   AddSettingsPageUIHandler(std::make_unique<PrivacySandboxHandler>());
   AddSettingsPageUIHandler(std::make_unique<SearchEnginesHandler>(profile));
+  AddSettingsPageUIHandler(
+      std::make_unique<HeliumServicesSchemaHandler>(profile));
   AddSettingsPageUIHandler(std::make_unique<SecureDnsHandler>());
   AddSettingsPageUIHandler(std::make_unique<SiteSettingsHandler>(profile));
   AddSettingsPageUIHandler(std::make_unique<StartupPagesHandler>(web_ui));
