--- /dev/null
+++ b/components/helium_services/extension_ids.h
@@ -0,0 +1,22 @@
+// Copyright 2025 The Helium Authors
+// You can use, redistribute, and/or modify this source code under
+// the terms of the GPL-3.0 license that can be found in the LICENSE file.
+
+#ifndef COMPONENTS_HELIUM_SERVICES_EXTENSION_IDS_H_
+#define COMPONENTS_HELIUM_SERVICES_EXTENSION_IDS_H_
+
+namespace helium {
+
+inline constexpr char kUBlockOriginWebstoreId[] =
+    "cjpalhdlnbpafiamejdnhcphjbkeiagm";
+
+inline constexpr char kUBlockOriginComponentId[] =
+    "blockjmkbacgjkknlgpkjjiijinjdanf";
+
+inline constexpr char kUblockOriginComponentIdHashed[] =
+    // SHA1(kUBlockOriginComponentId)
+    "70C2895ED505C532A80761F0928B821B103D0786";
+
+}  // namespace helium
+
+#endif  /* COMPONENTS_HELIUM_SERVICES_EXTENSION_IDS_H_ */
--- a/chrome/browser/extensions/component_loader.h
+++ b/chrome/browser/extensions/component_loader.h
@@ -218,6 +218,7 @@ class ComponentLoader : public KeyedServ
                                  const std::string& name_string,
                                  const std::string& description_string);
   void AddWebStoreApp();
+  void AddUBlock();
 
 #if BUILDFLAG(IS_CHROMEOS)
   void AddChromeApp();
--- a/chrome/browser/extensions/component_loader.cc
+++ b/chrome/browser/extensions/component_loader.cc
@@ -439,6 +439,21 @@ void ComponentLoader::AddWebStoreApp() {
       l10n_util::GetStringUTF8(IDS_WEBSTORE_APP_DESCRIPTION));
 }
 
+void ComponentLoader::AddUBlock() {
+  Add(IDR_UBLOCK_MANIFEST_JSON, base::FilePath(FILE_PATH_LITERAL("ublock")));
+
+  using namespace extension_l10n_util;
+  ComponentExtensionInfo& info = component_extensions_.back();
+  std::string error;
+  bool localized = LocalizeExtension(info.root_directory, &info.manifest,
+                      GzippedMessagesPermission::kAllowForTrustedSource,
+                      &error, true);
+
+  if (!localized) {
+    LOG(ERROR) << "Failed to localize uBlock: " << error;
+  }
+}
+
 #if BUILDFLAG(IS_CHROMEOS)
 void ComponentLoader::AddChromeApp() {
   AddWithNameAndDescription(
@@ -516,6 +531,7 @@ void ComponentLoader::AddDefaultComponen
 
   if (!skip_session_components) {
     AddWebStoreApp();
+    AddUBlock();
 #if BUILDFLAG(IS_CHROMEOS)
     AddChromeApp();
 #endif  // BUILDFLAG(IS_CHROMEOS)
--- a/chrome/browser/extensions/component_extensions_allowlist/allowlist.cc
+++ b/chrome/browser/extensions/component_extensions_allowlist/allowlist.cc
@@ -15,10 +15,11 @@
 #include "chrome/common/buildflags.h"
 #include "chrome/common/extensions/extension_constants.h"
 #include "chrome/grit/browser_resources.h"
+#include "components/helium_services/extension_ids.h"
 #include "chrome/grit/component_extension_resources.h"
 #include "extensions/buildflags/buildflags.h"
 #include "extensions/common/constants.h"
 #include "printing/buildflags/buildflags.h"
 
 #if BUILDFLAG(IS_CHROMEOS)
 #include "ash/keyboard/ui/grit/keyboard_resources.h"
@@ -36,6 +37,7 @@ bool IsComponentExtensionAllowlisted(con
       extension_misc::kGlicExtensionId,
       extension_misc::kInAppPaymentsSupportAppId,
       extension_misc::kPdfExtensionId,
+      helium::kUBlockOriginComponentId,
 #if BUILDFLAG(IS_CHROMEOS)
       extension_misc::kAssessmentAssistantExtensionId,
       extension_misc::kAccessibilityCommonExtensionId,
@@ -95,6 +97,7 @@ bool IsComponentExtensionAllowlisted(int
     case IDR_NETWORK_SPEECH_SYNTHESIS_MANIFEST_MV3:
     case IDR_READING_MODE_GDOCS_HELPER_MANIFEST:
     case IDR_WEBSTORE_MANIFEST:
+    case IDR_UBLOCK_MANIFEST_JSON:
 
 #if BUILDFLAG(IS_CHROMEOS)
     // Separate ChromeOS list, as it is quite large.
--- a/ui/base/resource/resource_bundle.cc
+++ b/ui/base/resource/resource_bundle.cc
@@ -761,7 +761,7 @@ base::RefCountedMemory* ResourceBundle::
 
   std::string_view data = GetRawDataResourceForScale(resource_id, scale_factor);
   if (data.empty())
-    return nullptr;
+    return new base::RefCountedString(); // fixes accessing "web_accessible_resources/empty" in u0
 
   if (net::GZipHeader::HasGZipHeader(base::as_byte_span(data)) ||
       HasBrotliHeader(data)) {
--- a/chrome/browser/resources/component_extension_resources.grd
+++ b/chrome/browser/resources/component_extension_resources.grd
@@ -21,6 +21,8 @@
 
       <include name="IDR_READING_MODE_GDOC_HELPER_CONTENT_JS" file="reading_mode_gdocs_helper/content.js" type="BINDATA" />
       <include name="IDR_READING_MODE_GDOC_HELPER_GDOCS_SCRIPT_JS" file="reading_mode_gdocs_helper/gdocs_script.js" type="BINDATA" />
 
+      <include name="IDR_UBLOCK_MANIFEST_JSON" file="../../../third_party/ublock/manifest.json" resource_path="ublock/manifest.json" type="BINDATA" />
+
       <if expr="_google_chrome">
         <then>
           <include name="IDR_GLIC_EXTENSION_MANIFEST" file="../../../internal/extensions/glic/manifest_internal.json" resource_path="glic_extension/manifest.json" type="BINDATA" />
