--- a/chrome/browser/ui/browser_ui_prefs.cc
+++ b/chrome/browser/ui/browser_ui_prefs.cc
@@ -70,7 +70,7 @@ void RegisterBrowserPrefs(PrefRegistrySi
 
   registry->RegisterBooleanPref(prefs::kHoverCardImagesEnabled, true);
 
-  registry->RegisterBooleanPref(prefs::kHoverCardMemoryUsageEnabled, true);
+  registry->RegisterBooleanPref(prefs::kHoverCardMemoryUsageEnabled, false);
 
   registry->RegisterBooleanPref(
       prefs::kHoverCardMemoryUsageDisableMigrationComplete, false);
@@ -167,7 +167,7 @@ void RegisterBrowserUserPrefs(user_prefs
 
   // We need to register the type of these preferences in order to query
   // them even though they're only typically controlled via policy.
-  registry->RegisterBooleanPref(policy::policy_prefs::kHideWebStoreIcon, false);
+  registry->RegisterBooleanPref(policy::policy_prefs::kHideWebStoreIcon, true);
   registry->RegisterBooleanPref(prefs::kSharedClipboardEnabled, true);
 
   registry->RegisterBooleanPref(prefs::kClickToCallEnabled, true);
