--- a/chrome/browser/ui/webui/settings/settings_localized_strings_provider.cc
+++ b/chrome/browser/ui/webui/settings/settings_localized_strings_provider.cc
@@ -36,7 +36,6 @@
 #include "chrome/browser/profiles/profile_attributes_storage.h"
 #include "chrome/browser/profiles/profile_manager.h"
 #include "chrome/browser/profiles/profile_shortcut_manager.h"
-#include "chrome/browser/regional_capabilities/regional_capabilities_service_factory.h"
 #include "chrome/browser/signin/account_consistency_mode_manager.h"
 #include "chrome/browser/signin/account_consistency_mode_manager_factory.h"
 #include "chrome/browser/signin/chrome_signin_client_factory.h"
@@ -93,7 +92,6 @@
 #include "components/plus_addresses/core/browser/plus_address_service.h"
 #include "components/plus_addresses/core/common/features.h"
 #include "components/prefs/pref_service.h"
-#include "components/regional_capabilities/regional_capabilities_service.h"
 #include "components/safe_browsing/core/common/features.h"
 #include "components/safe_browsing/core/common/hashprefix_realtime/hash_realtime_utils.h"
 #include "components/saved_tab_groups/public/features.h"
@@ -2918,14 +2916,9 @@ void AddSearchStrings(content::WebUIData
   html_source->AddString("searchExplanationLearnMoreURL",
                          chrome::kOmniboxLearnMoreURL);
 
-  regional_capabilities::RegionalCapabilitiesService* regional_capabilities =
-      regional_capabilities::RegionalCapabilitiesServiceFactory::GetForProfile(
-          profile);
   html_source->AddLocalizedString(
       "searchEnginesSettingsDialogSubtitle",
-      regional_capabilities->IsInEeaCountry()
-          ? IDS_SEARCH_ENGINE_CHOICE_SETTINGS_SUBTITLE
-          : IDS_SEARCH_ENGINE_CHOICE_SETTINGS_SUBTITLE_NON_EEA);
+      IDS_SEARCH_ENGINE_CHOICE_SETTINGS_SUBTITLE_HELIUM);
 
   html_source->AddLocalizedString(
       "saveGuestChoiceText", IDS_SEARCH_ENGINE_CHOICE_GUEST_SESSION_CHECKBOX);
--- a/components/regional_capabilities/regional_capabilities_service.cc
+++ b/components/regional_capabilities/regional_capabilities_service.cc
@@ -219,23 +219,7 @@ std::pair<CountryId, LoadedCountrySource
 // Pass `out_scope_check_outcome` to collect information about the outcome of
 // the program checks.
 Program CountryIdToProgram(const CountryId& country_id) {
-  static constexpr Program kCountryDerivedPrograms[] = {
-#if BUILDFLAG(IS_IOS)
-      // Only iOS can derive Taiyaki scope directly from the country.
-      Program::kTaiyaki,
-#endif
-
-      Program::kWaffle,
-  };
-
-  for (Program program : kCountryDerivedPrograms) {
-    if (IsInProgramRegion(program, country_id) &&
-        IsClientCompatibleWithProgram(program)) {
-      return program;
-    }
-  }
-
-  return Program::kDefault;
+  return Program::kWaffle;
 }
 
 std::optional<ProgramAndLocationMatch> GetProgramAndLocationMatch(
@@ -403,14 +387,13 @@ RegionalCapabilitiesService::GetRegional
 }
 
 bool RegionalCapabilitiesService::IsInSearchEngineChoiceScreenRegion() {
-  return GetChoiceScreenEligibilityConfig().has_value();
+  return true;
 }
 
 // static
 bool RegionalCapabilitiesService::IsInSearchEngineChoiceScreenRegion(
     const country_codes::CountryId& tested_country_id) {
-  return GetSettingsForProgram(CountryIdToProgram(tested_country_id))
-      .choice_screen_eligibility_config.has_value();
+  return true;
 }
 
 #if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_MAC) || BUILDFLAG(IS_LINUX)
@@ -530,14 +513,7 @@ RegionalCapabilitiesService::GetChoiceSc
 }
 
 bool RegionalCapabilitiesService::IsInEeaCountry() {
-  // Feature behaviour was directly based on the current country, as a
-  // decentralised way to express a concept we are now framing as "program
-  // settings". Here we check for the program reference directly as command line
-  // overrides may be setting a program with a separate country engine list
-  // override.
-  // TODO(crbug.com/328040066): Introduce granular program settings APIs and
-  // deprecate `IsInEeaCountry()` in favour of these.
-  return GetActiveProgramSettings().program == Program::kWaffle;
+  return true;
 }
 
 RegionalCapabilitiesService::Client&
--- a/components/search_engine_choice_strings.grdp
+++ b/components/search_engine_choice_strings.grdp
@@ -19,6 +19,10 @@
     <ph name="SEARCH_ENGINE_NAME">$1<ex>Google</ex></ph> is now your default search engine
   </message>
 
+  <message name="IDS_SEARCH_ENGINE_CHOICE_SETTINGS_SUBTITLE_HELIUM" desc="This is the body text of a dialog where the user can change their default search engine. This string explains that the list of search engines is randomized (not in order of recommendation).">
+    This list of search engines is shown in random order.
+  </message>
+
   <message name="IDS_SEARCH_ENGINE_CHOICE_SETTINGS_ENTRY_POINT_SUBTITLE" desc="This string is body text of a Chrome setting that lets a user select the search engine that they want to use. (A search engine provides answers to search queries; this is different from a browser, which displays web pages. A search engine powers many experiences within a browser.) The string explains how a search engine will impact their experience in Chrome. The “Learn more” link opens a help center article about the default search engine.">
     The search engine you choose will be used for features like searching from the address bar and from images on web pages.
   </message>
