Index: chromium_src/chrome/browser/ui/color/material_chrome_color_mixer.cc
===================================================================
--- a/chrome/browser/ui/color/material_chrome_color_mixer.cc
+++ b/chrome/browser/ui/color/material_chrome_color_mixer.cc
@@ -50,6 +50,9 @@ void ApplyDefaultChromeRefreshToolbarCol
 
 void AddMaterialChromeColorMixer(ui::ColorProvider* provider,
                                  const ui::ColorProviderKey& key) {
+  const bool dark_mode =
+      key.color_mode == ui::ColorProviderKey::ColorMode::kDark;
+
   // Adds the color recipes for browser UI colors (toolbar, bookmarks bar,
   // downloads bar etc). While both design systems continue to exist, the
   // material recipes are intended to leverage the existing chrome color mixers,
@@ -399,7 +402,8 @@ void AddMaterialChromeColorMixer(ui::Col
       ui::kColorSysOnSurfaceSubtle};
 
   // Toolbar colors.
-  mixer[kColorToolbar] = {ui::kColorSysBase};
+  mixer[kColorToolbar] = {dark_mode ? ui::kColorFrameActive
+                                    : ui::kColorSysBase};
   mixer[kColorToolbarButtonBackgroundHighlightedDefault] = {
       ui::kColorSysStateHoverOnSubtle};
   mixer[kColorToolbarButtonBorder] = {ui::kColorSysOutline};
@@ -411,7 +415,10 @@ void AddMaterialChromeColorMixer(ui::Col
       {ui::kColorSysStateDisabled}, {kColorToolbar})};
   mixer[kColorToolbarButtonIconPressed] = {kColorToolbarButtonIconHovered};
   mixer[kColorToolbarButtonText] = {ui::kColorSysOnSurfaceSecondary};
-  mixer[kColorToolbarContentAreaSeparator] = {ui::kColorSysSurfaceVariant};
+  mixer[kColorToolbarContentAreaSeparator] = {
+      dark_mode ? ui::GetResultingPaintColor({ui::kColorSysStateHoverOnSubtle},
+                                             {kColorToolbar})
+                : ui::kColorSysSurfaceVariant};
   mixer[kColorToolbarFeaturePromoHighlight] = {ui::kColorSysPrimary};
   mixer[kColorToolbarIconContainerBorder] = {ui::kColorSysNeutralOutline};
   mixer[kColorToolbarInkDropHover] = {ui::kColorSysStateHoverOnSubtle};
@@ -420,8 +427,11 @@ void AddMaterialChromeColorMixer(ui::Col
   mixer[kColorToolbarExtensionSeparatorDisabled] = {
       kColorToolbarButtonIconInactive};
   mixer[kColorToolbarSeparator] = {kColorToolbarSeparatorDefault};
+  const auto toolbar_separator_bg = dark_mode
+                                        ? ui::ColorTransform(ui::kColorSysBase)
+                                        : ui::ColorTransform(kColorToolbar);
   mixer[kColorToolbarSeparatorDefault] =
-      ui::AlphaBlend(kColorToolbarButtonIcon, kColorToolbar, 0x3A);
+      ui::AlphaBlend(kColorToolbarButtonIcon, toolbar_separator_bg, 0x3A);
   mixer[kColorToolbarText] = {kColorToolbarTextDefault};
   mixer[kColorToolbarTextDefault] = {ui::kColorSysOnSurfaceSecondary};
   mixer[kColorToolbarTextDisabled] = {kColorToolbarTextDisabledDefault};
Index: chromium_src/chrome/browser/ui/color/chrome_color_mixer.cc
===================================================================
--- a/chrome/browser/ui/color/chrome_color_mixer.cc
+++ b/chrome/browser/ui/color/chrome_color_mixer.cc
@@ -119,9 +119,13 @@ ui::ColorTransform GetToolbarTopSeparato
 // Apply updates to the Omnibox background color tokens per GM3 spec.
 void ApplyGM3OmniboxBackgroundColor(ui::ColorMixer& mixer,
                                     const ui::ColorProviderKey& key) {
+  const bool dark_mode =
+      key.color_mode == ui::ColorProviderKey::ColorMode::kDark;
   // Apply omnibox background color updates only to non-themed clients.
   if (!key.custom_theme) {
-    mixer[kColorLocationBarBackground] = {ui::kColorSysOmniboxContainer};
+    mixer[kColorLocationBarBackground] = {
+        dark_mode ? ui::ColorTransform(kColorOmniboxResultsBackground)
+                  : ui::ColorTransform(ui::kColorSysOmniboxContainer)};
     mixer[kColorLocationBarBackgroundHovered] =
         ui::GetResultingPaintColor(ui::kColorSysStateHoverBrightBlendProtection,
                                    kColorLocationBarBackground);
@@ -765,8 +769,10 @@ void AddChromeColorMixer(ui::ColorProvid
       ui::SetAlpha(kColorToolbarButtonIcon, gfx::kGoogleGreyAlpha500)};
   mixer[kColorToolbarButtonIconPressed] = {kColorToolbarButtonIconHovered};
   mixer[kColorToolbarButtonText] = ui::GetColorWithMaxContrast(kColorToolbar);
-  mixer[kColorToolbarContentAreaSeparator] =
-      ui::AlphaBlend(kColorToolbarButtonIcon, kColorToolbar, 0x3A);
+  mixer[kColorToolbarContentAreaSeparator] = {
+      dark_mode ? ui::GetResultingPaintColor({ui::kColorSysStateHoverOnSubtle},
+                                             {kColorToolbar})
+                : ui::kColorSysSurfaceVariant};
   mixer[kColorToolbarFeaturePromoHighlight] =
       ui::PickGoogleColor(ui::kColorAccent, kColorToolbar,
                           color_utils::kMinimumVisibleContrastRatio);
Index: chromium_src/ui/color/ui_color_mixer.cc
===================================================================
--- a/ui/color/ui_color_mixer.cc
+++ b/ui/color/ui_color_mixer.cc
@@ -31,7 +31,8 @@ void AddUiColorMixer(ColorProvider* prov
   mixer[kColorBadgeInCocoaMenuBackground] = {kColorBadgeBackground};
   mixer[kColorBadgeInCocoaMenuForeground] = {kColorBadgeForeground};
   mixer[kColorBubbleBackground] = {kColorPrimaryBackground};
-  mixer[kColorBubbleBorder] = {kColorSysSurfaceVariant};
+  mixer[kColorBubbleBorder] = {dark_mode ? ui::kColorSysStateHoverOnSubtle
+                                         : ui::kColorSysSurfaceVariant};
   mixer[kColorBubbleBorderShadowLarge] = {SetAlpha(kColorShadowBase, 0x1A)};
   mixer[kColorBubbleBorderShadowSmall] = {SetAlpha(kColorShadowBase, 0x33)};
   mixer[kColorBubbleFooterBackground] = {kColorSubtleEmphasisBackground};
Index: chromium_src/chrome/browser/ui/color/material_tab_strip_color_mixer.cc
===================================================================
--- a/chrome/browser/ui/color/material_tab_strip_color_mixer.cc
+++ b/chrome/browser/ui/color/material_tab_strip_color_mixer.cc
@@ -33,9 +33,9 @@ void AddMaterialTabStripColorMixer(ui::C
   ui::ColorMixer& mixer = provider->AddMixer();
   mixer[kColorDetachedTabBackgroundActiveFrameActive] = {
       dark_mode ? ui::kColorSysSurfaceVariant : ui::kColorSysBase};
-  mixer[kColorTabBackgroundActiveFrameActive] = {ui::kColorSysBase};
+  mixer[kColorTabBackgroundActiveFrameActive] = {kColorLocationBarBackground};
   mixer[kColorTabBackgroundActiveFrameInactive] = {
-      kColorTabBackgroundActiveFrameActive};
+      kColorLocationBarBackground};
 
   mixer[kColorTabBackgroundInactiveFrameActive] = {ui::kColorSysHeader};
   mixer[kColorTabBackgroundInactiveFrameInactive] = {
@@ -69,12 +69,15 @@ void AddMaterialTabStripColorMixer(ui::C
   mixer[kColorTabForegroundActiveFrameActive] = {ui::kColorSysOnSurface};
   mixer[kColorTabForegroundActiveFrameInactive] = {
       kColorTabForegroundActiveFrameActive};
-  mixer[kColorTabForegroundInactiveFrameActive] =
-      ui::BlendForMinContrast({ui::kColorSysOnSurfaceSecondary},
-                              {kColorTabBackgroundInactiveFrameActive});
-  mixer[kColorTabForegroundInactiveFrameInactive] =
-      ui::BlendForMinContrast({kColorTabForegroundInactiveFrameActive},
-                              {kColorTabBackgroundInactiveFrameInactive});
+  mixer[kColorTabForegroundInactiveFrameActive] = ui::BlendForMinContrast(
+      {ui::kColorSysOnSurfaceSecondary},
+      {kColorTabBackgroundInactiveFrameActive});
+  mixer[kColorTabForegroundInactiveFrameInactive] = ui::BlendForMinContrast(
+      {kColorTabForegroundInactiveFrameActive},
+      {kColorTabBackgroundInactiveFrameInactive});
+  mixer[kColorTabCloseButtonFocusRingInactive] = {
+      ui::kColorSysStateFocusRing};
+  mixer[kColorTabFocusRingInactive] = {ui::kColorSysStateFocusRing};
 
   /* WebUI Tab Strip colors. */
   mixer[kColorWebUiTabStripBackground] = {ui::kColorSysHeader};
@@ -98,10 +101,8 @@ void AddMaterialTabStripColorMixer(ui::C
       ui::kColorSysOnSurfaceSubtle};
   mixer[kColorNewTabButtonCRForegroundFrameInactive] = {
       ui::kColorSysOnSurfaceSubtle};
-  mixer[kColorNewTabButtonCRBackgroundFrameActive] = {
-      ui::kColorSysHeaderContainer};
-  mixer[kColorNewTabButtonCRBackgroundFrameInactive] = {
-      ui::kColorSysHeaderContainerInactive};
+  mixer[kColorNewTabButtonCRBackgroundFrameActive] = {kColorToolbar};
+  mixer[kColorNewTabButtonCRBackgroundFrameInactive] = {kColorToolbar};
 
   mixer[kColorTabSearchButtonCRForegroundFrameActive] = {
       ui::kColorSysOnSurfacePrimary};
Index: chromium_src/ui/color/sys_color_mixer.cc
===================================================================
--- a/ui/color/sys_color_mixer.cc
+++ b/ui/color/sys_color_mixer.cc
@@ -66,7 +66,7 @@ void AddThemedSysColorOverrides(ColorMix
 
   // Experimentation.
   mixer[kColorSysOmniboxContainer] = {dark_mode ? kColorRefSecondary15
-                                                : kColorSysSurface4};
+                                                : kColorSysSurface5};
 }
 
 // Overrides for the grayscale baseline case.
@@ -95,6 +95,10 @@ void AddGrayscaleSysColorOverrides(Color
   mixer[kColorSysOnHeaderPrimary] = {dark_mode ? kColorRefNeutral80
                                                : kColorRefNeutral40};
 
+  // Toned divider.
+  mixer[kColorSysSurfaceVariant] = {dark_mode ? kColorRefNeutral40
+                                              : kColorRefNeutral92};
+
   // States.
   mixer[kColorSysStateInactiveRing] = {
       dark_mode ? SetAlpha({kColorRefNeutral70}, 0x8C)
@@ -106,7 +110,7 @@ void AddGrayscaleSysColorOverrides(Color
 
   // Experimentation.
   mixer[kColorSysOmniboxContainer] = {dark_mode ? kColorRefNeutral15
-                                                : kColorRefNeutral94};
+                                                : kColorRefNeutral92};
 }
 
 }  // namespace
@@ -367,7 +371,7 @@ void AddSysColorMixer(ColorProvider* pro
 
   // Experimentation.
   mixer[kColorSysOmniboxContainer] = {dark_mode ? kColorRefNeutral15
-                                                : kColorSysSurface4};
+                                                : kColorSysSurface5};
 
   // Deprecated.
   // TODO(crbug.com/350783235): Remove remaining uses of these deprecated sys
