--- a/chrome/browser/ui/views/frame/layout/browser_view_layout_impl.cc
+++ b/chrome/browser/ui/views/frame/layout/browser_view_layout_impl.cc
@@ -294,7 +294,6 @@
     // correctly.
     background->SetVisible(true);
     background->SetPrimaryColor(ui::kColorFrameActive);
-    background->SetCorners(CustomCornersBackground::Corners());
   } else {
     // No need to paint otherwise.
     background->SetVisible(false);
--- a/chrome/browser/ui/views/frame/layout/browser_view_tabbed_layout_impl.cc
+++ b/chrome/browser/ui/views/frame/layout/browser_view_tabbed_layout_impl.cc
@@ -1319,25 +1319,6 @@
   } else {
     background->SetPrimaryColor(CustomCornersBackground::ToolbarTheme());
   }
-
-  // By default, this is just a flat background.
-  CustomCornersBackground::Corners corners;
-
-  // Rounded corners are drawn in vertical tab strip mode when not maximized or
-  // fullscreen.
-  if (delegate().ShouldDrawVerticalTabStrip() &&
-      delegate().GetBrowserWindowState() == WindowState::kNormal) {
-    corners.upper_trailing = background->GetWindowCorner(/*upper=*/true);
-    const bool vertical_tab_strip_reaches_top =
-        GetVerticalTabStripCollapsedState() !=
-            VerticalTabStripCollapsedState::kCollapsed ||
-        params.leading_exclusion.IsEmpty();
-    if (!vertical_tab_strip_reaches_top) {
-      corners.upper_leading = background->GetWindowCorner(/*upper=*/true);
-    }
-  }
-
-  background->SetCorners(corners);
 }
 
 void BrowserViewTabbedLayoutImpl::DoPostLayoutVisualAdjustments(
