--- a/chrome/browser/ui/cocoa/main_menu_builder.mm
+++ b/chrome/browser/ui/cocoa/main_menu_builder.mm
@@ -302,15 +302,6 @@ NSMenuItem* BuildViewMenu(NSApplication*
                   .command_id(IDC_TOGGLE_FULLSCREEN_TOOLBAR),
               Item(IDS_CONTEXT_MENU_SHOW_FULL_URLS)
                   .command_id(IDC_SHOW_FULL_URLS),
-              Item(IDS_CONTEXT_MENU_SHOW_GOOGLE_LENS_SHORTCUT)
-                  .command_id(IDC_SHOW_GOOGLE_LENS_SHORTCUT),
-              Item(IDS_CONTEXT_MENU_SHOW_AI_MODE_OMNIBOX_BUTTON)
-                  .command_id(IDC_SHOW_AI_MODE_OMNIBOX_BUTTON),
-              Item(IDS_CONTEXT_MENU_SHOW_SEARCH_TOOLS)
-                  .command_id(IDC_SHOW_SEARCH_TOOLS),
-              Item(IDS_SWITCH_TO_VERTICAL_TAB)
-                  .command_id(IDC_TOGGLE_VERTICAL_TABS)
-                  .remove_if(!tabs::IsVerticalTabsFeatureEnabled()),
               Item(IDS_CUSTOMIZE_TOUCH_BAR)
                   .tag(IDC_CUSTOMIZE_TOUCH_BAR)
                   .action(@selector(toggleTouchBarCustomizationPalette:))
@@ -536,7 +527,8 @@ NSMenuItem* BuildTabMenu(NSApplication*
 
   // clang-format off
   NSMenuItem* item =
-      Item(IDS_TAB_MENU_MAC)
+    // Hopefully translates to "Tabs" in all langs
+      Item(IDS_SEARCH_ENGINES_STARTER_PACK_TABS_NAME)
           .tag(IDC_TAB_MENU)
           .submenu({
               Item(is_rtl ? IDS_TAB_CXMENU_NEWTABTOLEFT
@@ -595,26 +587,9 @@ NSMenuItem* BuildHelpMenu(NSApplication*
     return nil;
   }
 
-  // clang-format off
-  std::vector<Item> items = {};
-#if BUILDFLAG(GOOGLE_CHROME_BRANDING)
-  items.push_back(Item(IDS_FEEDBACK_MAC).command_id(IDC_FEEDBACK));
-  bool is_profile_loaded =
-      g_browser_process && g_browser_process->profile_manager() &&
-      g_browser_process->profile_manager()->GetLastUsedProfileIfLoaded();
-  if (is_profile_loaded && feedback::ReportUnsafeSiteDialog::IsEnabled(
-                               *(g_browser_process->profile_manager()
-                                     ->GetLastUsedProfileIfLoaded()))) {
-    items.push_back(Item(IDS_REPORT_UNSAFE_SITE)
-                        .command_id(IDC_REPORT_UNSAFE_SITE));
-  }
-#endif
-  items.push_back(Item(IDS_HELP_MAC)
-                      .string_format_1(product_name)
-                      .command_id(IDC_HELP_PAGE_VIA_MENU));
   NSMenuItem* item =
       Item(IDS_HELP_MENU_MAC)
-          .submenu(items)
+          .submenu({})
           .Build();
   // clang-format on
 
@@ -637,11 +612,11 @@ NSMenu* BuildMainMenu(NSApplication* nsa
            &BuildFileMenu,
            &BuildEditMenu,
            &BuildViewMenu,
-           &BuildHistoryMenu,
-           &BuildBookmarksMenu,
+           &BuildTabMenu,
            &BuildGroupsMenu,
+           &BuildBookmarksMenu,
+           &BuildHistoryMenu,
            &BuildPeopleMenu,
-           &BuildTabMenu,
            &BuildWindowMenu,
            &BuildHelpMenu,
        }) {
